---
title: "From gprofiler2 to Cytoscape network"
author: Astrid Deschênes and Pascal Belleau
output:
  BiocStyle::html_document:
    toc: true
bibliography: gprofiler2cytoscape.bibtex
vignette: >
  %\VignetteIndexEntry{From gprofiler2 to Cytoscape network}
  %\VignettePackage{gprofiler2cytoscape}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r style, echo = FALSE, results = 'hide', warning=FALSE, message=FALSE}
BiocStyle::markdown()

suppressPackageStartupMessages({
  library(knitr)
  library(gprofiler2cytoscape)
})

set.seed(1214)
```

<br />
**Package**: `r Rpackage("gprofiler2cytoscape")`<br />
**Authors**: `r packageDescription("gprofiler2cytoscape")[["Author"]]`<br />
**Version**: `r packageDescription("gprofiler2cytoscape")$Version`<br />
**Compiled date**: `r Sys.Date()`<br />
**License**: `r packageDescription("gprofiler2cytoscape")[["License"]]`<br />


# Licensing 

The `r Githubpkg("adeschen/gprofiler2cytoscape")` package and the underlying 
`r Githubpkg("adeschen/gprofiler2cytoscape")` code are distributed under the 
Artistic license 2.0. You are free to use and redistribute this software. 


# Citing

If you use this package for a publication, we would ask you to cite the 
following:

> Astrid Deschênes and Pascal Belleau (2021). gprofiler2cytoscape: From 
  gprofiler2 to Cytoscape network. 
  https://github.com/adeschen/gprofiler2cytoscape,
  https://adeschen.github.io/gprofiler2cytoscape/.


# Introduction

Biology studies that use high-throughput technologies, such as RNA sequencing 
or microarray platform, usually end up generating large amount of data such 
as gene lists. For instance, RNA sequencing is frequently uses to discover 
genes that have differential expression between two biological states 
(such as healthy versus diseased). Those studies can make use of statistical 
tools that enable positioning those novel gene lists against previous 
knowledge.

Functional enrichment analysis tools such as _Enrichr_ [@Kuleshov2016] and 
_g:Profiler_ [@Raudvere2019] specialize in positioning novel findings against 
well curated data sources of biological processes and pathways. Most 
specifically, those tools identify functionally coherent gene sets that are 
statistically over- (or under-) represented in a gene list. The 
traditional output of enrichment analysis tool is a table containing the
significant gene sets with their associated statistics. While those results
are extremely useful, their interpretation is not always easy.

The _g:Profiler_ enrichment  analysis tool is web based as the particularity of
being accompanied by the CRAN package _gprofiler2_ [@Kolberg2020]. The 
_gprofiler2_ package enables researchers to incorporate functional 
enrichment analysis into automated analysis pipelines written in R. This 
greatly facilitates research reproducibility. 


The `r Githubpkg("adeschen/gprofiler2cytoscape")` package enables the 
visualization of enrichment results generated by _gprofiler2_ as 
customisable _Cytoscape_ network [@PaulShannon2003].

TODO

# Installation

To install the latest version accessible on the 
`r Githubpkg("adeschen/gprofiler2cytoscape")` Github Website, the 
`r CRANpkg("devtools")` package is required.

```{r installDemo01, eval=FALSE}
## Load required package
library(devtools)
## Install the latest version of gprofiler2cytoscape
devtools::install_github('adeschen/gprofiler2cytoscape')
```

It is also possible to install an official release. The list of available 
releases is posted on the [gprofiler2cytoscape Release Website](https://github.com/adeschen/gprofiler2cytoscape/releases "gprofiler2cytoscape Release Site").

```{r installDemo02, eval=FALSE}
## Load required package
library(devtools)
## Install the version v0.1 of gprofiler2cytoscape
## using 'ref' parameter
devtools::install_github('adeschen/gprofiler2cytoscape', ref = "v0.1")
```

TODO

# Session info

Here is the output of sessionInfo() on the system on which this document 
was compiled:

```{r sessionInfo, echo=FALSE}
sessionInfo()
```


# References


