---
title: "From gprofiler2 to Cytoscape network"
author: Astrid Deschênes and Pascal Belleau
output:
  BiocStyle::html_document:
    toc: true
pkgdown:
  as_is: true
bibliography: gprofiler2cytoscape.bibtex
vignette: >
  %\VignetteIndexEntry{From gprofiler2 to Cytoscape network}
  %\VignettePackage{gprofiler2cytoscape}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r style, echo = FALSE, results = 'hide', warning=FALSE, message=FALSE}
BiocStyle::markdown()

suppressPackageStartupMessages({
  library(knitr)
  library(gprofiler2cytoscape)
})

set.seed(1214)
```

<br />
**Package**: `r Rpackage("gprofiler2cytoscape")`<br />
**Authors**: `r packageDescription("gprofiler2cytoscape")[["Author"]]`<br />
**Version**: `r packageDescription("gprofiler2cytoscape")$Version`<br />
**Compiled date**: `r Sys.Date()`<br />
**License**: `r packageDescription("gprofiler2cytoscape")[["License"]]`<br />


# Licensing 

The `r Githubpkg("adeschen/gprofiler2cytoscape")` package and the underlying 
`r Githubpkg("adeschen/gprofiler2cytoscape")` code are distributed under the 
Artistic license 2.0. You are free to use and redistribute this software. 

<br>
<br>

# Citing

If you use this package for a publication, we would ask you to cite the 
following:

> Astrid Deschênes and Pascal Belleau (2021). gprofiler2cytoscape: From 
  gprofiler2 to Cytoscape network. 
  https://github.com/adeschen/gprofiler2cytoscape,
  https://adeschen.github.io/gprofiler2cytoscape/.

<br>
<br>

# Introduction

Biology studies that use high-throughput technologies, such as RNA sequencing 
or microarray platform, usually end up generating large amount of data such 
as gene lists. For instance, RNA sequencing is frequently uses to discover 
genes that have differential expression between two biological states 
(such as healthy versus diseased). Those studies can make use of statistical 
tools that enable positioning those novel gene lists against previous 
knowledge.

Functional enrichment analysis tools, such as _Enrichr_ [@Kuleshov2016] and 
_DAVID_ [@Dennis2003], specialize in positioning novel findings against 
well curated data sources of biological processes and pathways. Most 
specifically, those tools identify functionally coherent gene sets that are 
statistically over- (or under-) represented in a gene list. The 
traditional output of enrichment analysis tool is a table containing the
significant gene sets with their associated statistics. While those results
are extremely useful, their interpretation is not always easy. In that 
respect, visual representation of the results as network could be handy. 

Biological network models are visual representation of various biological 
interacting elements which are based on mathematical graphs. In those 
networks, the biological elements are generally represented by nodes while 
the interactions and relationships are represented by edges. One of the 
wildly used network tool in the quantitative biology community is the 
open source software _Cytoscape_. In addition of biological data 
visualization and network
analysis, _Cytoscape_ can be expended through the use of 
specialized plug-ins such as _BiNGO_ that calculates over-represented GO terms 
in a network [@Maere2005] or _CentiScaPe_ that identifies relevant network 
nodes [@Scardoni2009]. 

The _g:Profiler_ enrichment  analysis tool [@Raudvere2019] is web based as 
the particularity of being accompanied by the CRAN package 
_gprofiler2_ [@Kolberg2020]. The _gprofiler2_ package enables researchers to 
incorporate functional enrichment analysis into automated analysis pipelines 
written in R. This greatly facilitates research reproducibility. 

The **_gprofiler2cytoscape_** package enables the 
visualization of enrichment results generated by _gprofiler2_ as 
customisable _Cytoscape_ network [@PaulShannon2003]. In the biological 
networks generated by `r Githubpkg("adeschen/gprofiler2cytoscape")`, both 
gene datasets (GO terms/pathways/protein complexes) and genes associated 
to the datasets are represented as nodes. While the edges connect each gene 
to its dataset(s). Only genes present in the query used for the
enrichment analysis are shown.

```{r graphDemo01, echo = FALSE, fig.align="left", fig.cap="Demo a biologocial network where GO terms and genes are presented as nodes while edges connect each gene to its dataset(s).", out.width = '90%'}
knitr::include_graphics("demo01.jpeg")
```

The **_gprofiler2cytoscape_** package 
offers the option to generate a network for only a portion of the 
significant terms by selecting the source or by providing a 
specific list of terms.

Once the network is created, the user can personalize the visual attributes 
and integrate external information such as expression profiles, phenotypes 
and other molecular states. The user can also perform network analysis.

<br>
<br>

# Installation

To install the latest version accessible on the 
`r Githubpkg("adeschen/gprofiler2cytoscape")` Github Website, the 
`r CRANpkg("devtools")` package is required.

```{r installDemo01, eval=FALSE}
## Install the latest version of gprofiler2cytoscape
library(devtools)
devtools::install_github('adeschen/gprofiler2cytoscape')
```

It is also possible to install an official release. The list of available 
releases is posted on the [gprofiler2cytoscape Release Website](https://github.com/adeschen/gprofiler2cytoscape/releases "gprofiler2cytoscape Release Site").

```{r installDemo02, eval=FALSE}
## Install the version v0.1 of gprofiler2cytoscape
## using 'ref' parameter
library(devtools)
devtools::install_github('adeschen/gprofiler2cytoscape', ref = "v0.0.4")
```

<br>
<br>


# Transforming gprofiler2 enrichment results to Cytoscape network


```{r graphListToGraph01, echo = FALSE, fig.align="left", fig.cap="From an enrichment list to a Cytoscape network", out.width = '100%'}
knitr::include_graphics("FromListToGraph_v01.jpg")
```

The following workflow gives an overview of the capabilities of 
`r Githubpkg("adeschen/gprofiler2cytoscape")` 


The key steps for the workflow are:

Step                              | Function                    
--------------------------------- | ---------------------------------------------
 Run an Enrichment analysis       | `gprofiler2::gost()`
 Start Cytoscape                  |  outside R   
 Create Cytoscape network         | `createNetwork()`

The `package::function()` notation is used for functions from other packages.

## Run an enrichment analysis 

The first step consists in running an enrichment analysis with 
`r CRANpkg("gprofiler2")` package. The output of the _gprofiler2::gost()_ 
is a _list_ and should be saved. 


```{r gprofiler, echo = FALSE}
library(gprofiler2)

```

## Start Cytoscape

[Cytoscape](https://cytoscape.org/) is an open source software for 
visualizing networks. It enables network integration with any type of attribute
data. The Cytoscape software 
is available at the [Cytoscape website](https://cytoscape.org/).

```{r cytoscapeLogo01, echo = FALSE, fig.align="center", fig.cap="Cytoscape software logo.", out.width = '100%'}
knitr::include_graphics("cy3sticker.png")
```

The Cytoscape network generated by 
`r Githubpkg("adeschen/gprofiler2cytoscape")` 
will be automatically loaded into the [Cytoscape](https://cytoscape.org/) 
software when the application is running.

If the application is not running, a CX JSON file will be created. The file 
can then be loaded manually into the [Cytoscape](https://cytoscape.org/) 
software.

## Create Cytoscape network 

TODO

```{r runCreateNetwork, echo = TRUE, eval=FALSE}
## Load demo gprofiler2 enrichment result
data(demoGOST)

## Create network for Gene Ontology - Molecular Function related results
## The 'removeRoot=TRUE' parameter removes the root term from the network
createNetwork(gostObject=demoGOST, source="GO:MF", removeRoot=FALSE,
    title="GO Molecular Function Network")
```

<br>
<br>

# Session info

Here is the output of sessionInfo() on the system on which this document 
was compiled:

```{r sessionInfo, echo=FALSE}
sessionInfo()
```


<br>
<br>

# References


