<!DOCTYPE html>
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="pandoc">
<meta http-equiv="X-UA-Compatible" content="IE=EDGE">
<meta name="author" content="Astrid Deschênes, Pascal Belleau, Robert L Faure and Maria J Fernandes">
<meta name="date" content="2024-04-09">
<title>From functional enrichment results to biological networks</title>
<script src="enrichViewNet_files/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1">
<link href="enrichViewNet_files/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet">
<script src="enrichViewNet_files/bootstrap-3.3.5/js/bootstrap.min.js"></script><script src="enrichViewNet_files/bootstrap-3.3.5/shim/html5shiv.min.js"></script><script src="enrichViewNet_files/bootstrap-3.3.5/shim/respond.min.js"></script><style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       </style>
<script src="enrichViewNet_files/navigation-1.1/tabsets.js"></script><style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>
<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script><style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>
<link rel="stylesheet" href="/__w/_temp/Library/BiocStyle/resources/html/bioconductor.css" type="text/css">
<style type="text/css">
.main-container {
  max-width: 828px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
</style>
<!-- tabsets --><style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>
<!-- code folding --><script>
function toggle_visibility(id1) {
  var e = document.getElementById(id1);
  e.style.display = ((e.style.display!="none") ? "none" : "block");
}
</script>
</head>
<body>


<div class="container-fluid main-container">




<div id="header">



<h1 class="title toc-ignore">From functional enrichment results to biological networks</h1>
<p class="author-name">Astrid Deschênes, Pascal Belleau, Robert L Faure and Maria J Fernandes</p>
<h4 class="date">9 April 2024</h4>

</div>

<h1>Contents</h1>
<div id="TOC">
<ul>
<li><a href="#licensing" id="toc-licensing"><span class="toc-section-number">1</span> Licensing</a></li>
<li><a href="#citing" id="toc-citing"><span class="toc-section-number">2</span> Citing</a></li>
<li><a href="#introduction" id="toc-introduction"><span class="toc-section-number">3</span> Introduction</a></li>
<li><a href="#installation" id="toc-installation"><span class="toc-section-number">4</span> Installation</a></li>
<li><a href="#general-workflow" id="toc-general-workflow"><span class="toc-section-number">5</span> General workflow</a></li>
<li>
<a href="#transforming-enrichment-results-into-a-gene-term-network-loadable-in-cytoscape" id="toc-transforming-enrichment-results-into-a-gene-term-network-loadable-in-cytoscape"><span class="toc-section-number">6</span> Transforming enrichment results into a gene-term network loadable in Cytoscape</a>
<ul>
<li><a href="#run-an-enrichment-analysis" id="toc-run-an-enrichment-analysis"><span class="toc-section-number">6.1</span> Run an enrichment analysis</a></li>
<li><a href="#start-cytoscape" id="toc-start-cytoscape"><span class="toc-section-number">6.2</span> Start Cytoscape</a></li>
<li><a href="#create-a-gene-term-network" id="toc-create-a-gene-term-network"><span class="toc-section-number">6.3</span> Create a gene-term network</a></li>
</ul>
</li>
<li>
<a href="#transforming-enrichment-results-into-an-enrichment-map" id="toc-transforming-enrichment-results-into-an-enrichment-map"><span class="toc-section-number">7</span> Transforming enrichment results into an enrichment map</a>
<ul>
<li><a href="#run-an-enrichment-analysis-1" id="toc-run-an-enrichment-analysis-1"><span class="toc-section-number">7.1</span> Run an enrichment analysis</a></li>
<li>
<a href="#create-an-enrichment-map" id="toc-create-an-enrichment-map"><span class="toc-section-number">7.2</span> Create an enrichment map</a>
<ul>
<li><a href="#using-list-of-term-ids" id="toc-using-list-of-term-ids"><span class="toc-section-number">7.2.1</span> Using list of term IDs</a></li>
<li><a href="#effect-of-seed-value" id="toc-effect-of-seed-value"><span class="toc-section-number">7.2.2</span> Effect of <em>seed</em> value</a></li>
<li><a href="#enrichment-map-customization" id="toc-enrichment-map-customization"><span class="toc-section-number">7.2.3</span> Enrichment map customization</a></li>
</ul>
</li>
</ul>
</li>
<li>
<a href="#transforming-enrichment-results-into-an-enrichment-map-with-groups-from-different-enrichment-analyses" id="toc-transforming-enrichment-results-into-an-enrichment-map-with-groups-from-different-enrichment-analyses"><span class="toc-section-number">8</span> Transforming enrichment results into an enrichment map with groups from different enrichment analyses</a>
<ul>
<li><a href="#create-an-enrichment-map-using-multiple-enrichment-analyses" id="toc-create-an-enrichment-map-using-multiple-enrichment-analyses"><span class="toc-section-number">8.1</span> Create an enrichment map using multiple enrichment analyses</a></li>
<li><a href="#effect-of-seed-value-1" id="toc-effect-of-seed-value-1"><span class="toc-section-number">8.2</span> Effect of <em>seed</em> value</a></li>
<li><a href="#enrichment-map-customization-1" id="toc-enrichment-map-customization-1"><span class="toc-section-number">8.3</span> Enrichment map customization</a></li>
</ul>
</li>
<li>
<a href="#transforming-enrichment-results-into-an-enrichment-map-with-groups-from-same-enrichment-analysis" id="toc-transforming-enrichment-results-into-an-enrichment-map-with-groups-from-same-enrichment-analysis"><span class="toc-section-number">9</span> Transforming enrichment results into an enrichment map with groups from same enrichment analysis</a>
<ul>
<li><a href="#create-an-enrichment-map-using-subsections-of-one-enrichment-analysis" id="toc-create-an-enrichment-map-using-subsections-of-one-enrichment-analysis"><span class="toc-section-number">9.1</span> Create an enrichment map using subsections of one enrichment analysis</a></li>
</ul>
</li>
<li><a href="#acknowledgments" id="toc-acknowledgments"><span class="toc-section-number">10</span> Acknowledgments</a></li>
<li><a href="#session-info" id="toc-session-info"><span class="toc-section-number">11</span> Session info</a></li>
<li><a href="#references" id="toc-references">References</a></li>
</ul>
</div>

<p><br><strong>Package</strong>: <em>enrichViewNet</em><br><strong>Authors</strong>: Astrid Deschênes [aut, cre] (<a href="https://orcid.org/0000-0001-7846-6749" class="external-link uri">https://orcid.org/0000-0001-7846-6749</a>),
Pascal Belleau [aut] (<a href="https://orcid.org/0000-0002-0802-1071" class="external-link uri">https://orcid.org/0000-0002-0802-1071</a>),
Robert L. Faure [aut] (<a href="https://orcid.org/0000-0003-1798-4723" class="external-link uri">https://orcid.org/0000-0003-1798-4723</a>),
Maria J. Fernandes [aut] (<a href="https://orcid.org/0000-0002-3973-025X" class="external-link uri">https://orcid.org/0000-0002-3973-025X</a>),
Alexander Krasnitz [aut],
David A. Tuveson [aut] (<a href="https://orcid.org/0000-0002-8017-2712" class="external-link uri">https://orcid.org/0000-0002-8017-2712</a>)<br><strong>Version</strong>: 1.1.1<br><strong>Compiled date</strong>: 2024-04-09<br><strong>License</strong>: Artistic-2.0<br></p>
<div class="section level2" number="1">
<h2 id="licensing">
<span class="header-section-number">1</span> Licensing<a class="anchor" aria-label="anchor" href="#licensing"></a>
</h2>
<p>The <em><a href="https://bioconductor.org/packages/3.19/enrichViewNet" class="external-link">enrichViewNet</a></em> package and the underlying
<em><a href="https://bioconductor.org/packages/3.19/enrichViewNet" class="external-link">enrichViewNet</a></em> code are distributed under the
Artistic license 2.0. You are free to use and redistribute this software.</p>
<p><br><br></p>
</div>
<div class="section level2" number="2">
<h2 id="citing">
<span class="header-section-number">2</span> Citing<a class="anchor" aria-label="anchor" href="#citing"></a>
</h2>
<p>If you use this package for a publication, we would ask you to cite the
following:</p>
<blockquote>
<p>Deschênes A, Belleau P, Faure RL, Fernandes MJ, Krasnitz A,
Tuveson DA (2023).
enrichViewNet: From functional enrichment results to biological networks.
<a href="doi:10.18129/B9.bioc.enrichViewNet" class="uri">doi:10.18129/B9.bioc.enrichViewNet</a>,
<a href="https://bioconductor.org/packages/enrichViewNet" class="external-link uri">https://bioconductor.org/packages/enrichViewNet</a>.</p>
</blockquote>
<p><br><br></p>
</div>
<div class="section level2" number="3">
<h2 id="introduction">
<span class="header-section-number">3</span> Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>High-throughput technologies are routinely used in basic and applied
research and are key drivers of scientific discovery. A major challenge
in using these experimental approaches is the analysis of the large amount
of data generated. These include lists of proteins or genes generated by
mass spectrometry, single-cell RNA sequencing and/or microarray analysis,
respectively. There is thus a need for robust bioinformatic and statistical
tools that can analyze these large datasets and display the data in the form
of networks that illustrate the biological and conceptual links with findings
in the literature. This gap has been partially addressed by several
bioinformatic tools that perform enrichment analysis of the data and/or
present it in the form of networks.</p>
<p>Functional enrichment analysis tools, such as <em>Enrichr</em> <span class="citation">(<a href="#ref-Kuleshov2016">Kuleshov et al. 2016</a>)</span> and
<em>DAVID</em> <span class="citation">(<a href="#ref-Dennis2003">Dennis et al. 2003</a>)</span>, are specialized in positioning novel findings against
well curated data sources of biological processes and pathways.
Most specifically, those tools identify functional gene sets that
are statistically over- (or under-) represented in a gene list (functional
enrichment). The traditional output of a significant enrichment analysis
tool is a table containing
the significant gene sets with their associated statistics. While those
results are extremely useful, their interpretation is challenging. The visual
representation of these results as a network can greatly facilitate the
interpretation of the data.</p>
<p>Biological network models are visual representations of various biological
interacting elements which are based on mathematical graphs. In those
networks, the biological elements are generally represented by nodes while
the interactions and relationships are represented by edges. One of the
widely used network tools in the quantitative biology community is the
open source software <em>Cytoscape</em> <span class="citation">(<a href="#ref-PaulShannon2003">Shannon et al. 2003</a>)</span>. In addition of
biological data visualization and network
analysis, <em>Cytoscape</em> can be expended through the use of
specialized plug-ins such as <em>BiNGO</em> that calculates over-represented GO terms
in a network <span class="citation">(<a href="#ref-Maere2005">Maere, Heymans, and Kuiper 2005</a>)</span> or <em>CentiScaPe</em> that identifies relevant network
nodes <span class="citation">(<a href="#ref-Scardoni2009">Scardoni, Petterlini, and Laudanna 2009</a>)</span>.</p>
<p>The <em>g:Profiler</em> enrichment analysis tool <span class="citation">(<a href="#ref-Raudvere2019">Raudvere et al. 2019</a>)</span> is web based and has
the particularity of being accompanied by the CRAN package
<em>gprofiler2</em> <span class="citation">(<a href="#ref-Kolberg2020">Kolberg et al. 2020</a>)</span>. The <em>gprofiler2</em> package gives the opportunity to
researchers to incorporate functional enrichment analysis into automated
analysis pipelines written in R. This greatly facilitates research
reproducibility.</p>
<p>The <strong>enrichViewNet</strong> package enables the visualization
of functional enrichment results as network graphs. Visualization of
enriched terms aims to
facilitate the analyses of complex results.
Compared to popular enrichment visualization graphs such as bar plots and
dot plots, network graphs unveil the connection between the
terms as significant terms often share one or multiple genes. Moreover, the
<strong>enrichViewNet</strong> package takes advantage of a powerful network
visualization tool which is <em>Cytoscape</em>. By doing so, all the
functionalities of this mature software can be used to personalize
and analyze the enrichment networks.</p>
<p>First, the <strong>enrichViewNet</strong> package enables the
visualization of enrichment results, in a format corresponding to the one
generated by <em>gprofiler2</em>, as a
customizable <em>Cytoscape</em> network <span class="citation">(<a href="#ref-PaulShannon2003">Shannon et al. 2003</a>)</span>. In the biological
networks generated by <strong>enrichViewNet</strong>, both
gene datasets (GO terms/pathways/protein complexes) and genes associated
to the datasets are represented as nodes. While the edges connect each gene
to its dataset(s). Only genes present in the query used for the
enrichment analysis are shown.</p>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:graphDemo01"></span>
<img src="demo01.jpeg" alt="A network where significant GO terms and genes are presented as nodes while edges connect each gene to its associated term(s)." width="90%"><p class="caption">
Figure 1: <span class="caption-title">A network where significant GO terms and genes are presented as nodes while edges connect each gene to its associated term(s)</span><br></p>
</div>
<p>The <strong>enrichViewNet</strong> package
offers the option to generate a network for only a portion of the
significant terms by selecting the source or by providing a
specific list of terms.Once the network is created, the user can
personalize the visual attributes
and integrate external information such as expression profiles, phenotypes
and other molecular states. The user can also perform network analysis.</p>
<p>In addition, the <strong>enrichViewNet</strong> package also provides the option to
create enrichment maps from functional enrichment results.
The enrichment maps have been introduced in the Bioconductor
<em>enrichplot</em> package <span class="citation">(<a href="#ref-Yu2022">Yu 2022</a>)</span>.
Enrichment maps enable the visualization of enriched terms into a network
with edges connecting overlapping genes. Thus, enriched terms with overlapping
genes cluster together. This type of graphs facilitate the
identification of functional modules.</p>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:graphDemo02"></span>
<img src="demo_KEGG_emap_v01.jpeg" alt="An enrichment map using significant Kegg terms. Edges connecting terms with overlapping genes." width="95%"><p class="caption">
Figure 2: <span class="caption-title">An enrichment map using significant Kegg terms</span><br>Edges connecting terms with overlapping genes.
</p>
</div>
<p><strong>enrichViewNet</strong> has been submitted to
<a href="https://www.bioconductor.org/" class="external-link">Bioconductor</a> to aid researchers in carrying
out reproducible network analyses using functional enrichment results.</p>
<p><br><br></p>
</div>
<div class="section level2" number="4">
<h2 id="installation">
<span class="header-section-number">4</span> Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>To install this package
from <a href="https://bioconductor.org" class="external-link">Bioconductor</a>, start R
(version 4.3 or later) and enter:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>        <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"enrichViewNet"</span><span class="op">)</span></span></code></pre></div>
<p><br><br></p>
</div>
<div class="section level2" number="5">
<h2 id="general-workflow">
<span class="header-section-number">5</span> General workflow<a class="anchor" aria-label="anchor" href="#general-workflow"></a>
</h2>
<p>The following workflow gives an overview of the capabilities of
<strong>enrichViewNet</strong>:</p>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:graphWorkflow"></span>
<img src="Figure_enrichViewNet_workflow_v05.jpg" alt="The enrichViewNet general workflow" width="100%"><p class="caption">
Figure 3: <span class="caption-title">The enrichViewNet general workflow</span><br></p>
</div>
<p>The principal input of <strong>enrichViewNet</strong> is a functional enrichment
result in a format identical to the one generated by the CRAN
<em>gprofiler2</em> package.</p>
<p>From an enrichment result, the <strong>enrichViewNet</strong> offers two options:</p>
<ul>
<li>generate a gene-term network that can be loaded in <em>Cytoscape</em> software</li>
<li>generate a enrichment map</li>
</ul>
<p>For the gene-term network, the installation of <em>Cytoscape</em> software is
highly recommended.</p>
<p><br><br></p>
</div>
<div class="section level2" number="6">
<h2 id="transforming-enrichment-results-into-a-gene-term-network-loadable-in-cytoscape">
<span class="header-section-number">6</span> Transforming enrichment results into a gene-term network loadable in Cytoscape<a class="anchor" aria-label="anchor" href="#transforming-enrichment-results-into-a-gene-term-network-loadable-in-cytoscape"></a>
</h2>
<p>The following workflow gives an overview of the steps associated to the
creation of an gene-term network loadable in <em>Cytoscape</em>.</p>
<div class="figure" style="text-align: left">
<span style="display:block;" id="fig:graphListToGraph01"></span>
<img src="FromListToGraph_v02.jpg" alt="From an enrichment list to a Cytoscape network" width="100%"><p class="caption">
Figure 4: <span class="caption-title">From an enrichment list to a Cytoscape network</span><br></p>
</div>
<p>The key steps for the workflow are:</p>
<div class="horizontal-scroll"><table class="table">
<colgroup>
<col width="42%">
<col width="57%">
</colgroup>
<thead><tr class="header">
<th>Step</th>
<th>Function</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>Run an enrichment analysis</td>
<td><code><a href="https://rdrr.io/pkg/gprofiler2/man/gost.html" class="external-link">gprofiler2::gost()</a></code></td>
</tr>
<tr class="even">
<td>Start <em>Cytoscape</em>
</td>
<td>outside R</td>
</tr>
<tr class="odd">
<td>Create a gene-term network</td>
<td><code><a href="../reference/createNetwork.html">createNetwork()</a></code></td>
</tr>
</tbody>
</table></div>
<p>The <code>package::function()</code> notation is used for functions from other packages.</p>
<p><br></p>
<div class="section level3" number="6.1">
<h3 id="run-an-enrichment-analysis">
<span class="header-section-number">6.1</span> Run an enrichment analysis<a class="anchor" aria-label="anchor" href="#run-an-enrichment-analysis"></a>
</h3>
<p>The first step consists in running an enrichment analysis with
<em><a href="https://CRAN.R-project.org/package=gprofiler2" class="external-link">gprofiler2</a></em> package. The output of the <em>gprofiler2::gost()</em>
is a <em>list</em> and should be saved.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Required library</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">gprofiler2</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## The dataset of differentially expressed genes done between </span></span>
<span><span class="co">## napabucasin treated and DMSO control parental (Froeling et al 2019)</span></span>
<span><span class="co">## All genes tested are present</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"parentalNapaVsDMSODEG"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Retain significant results </span></span>
<span><span class="co">## (absolute fold change superior to 1 and adjusted p-value inferior to 0.05)</span></span>
<span><span class="va">retained</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="va">parentalNapaVsDMSODEG</span><span class="op">$</span><span class="va">log2FoldChange</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">1</span> <span class="op">&amp;</span> </span>
<span>                            <span class="va">parentalNapaVsDMSODEG</span><span class="op">$</span><span class="va">padj</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span></span>
<span><span class="va">signRes</span> <span class="op">&lt;-</span>  <span class="va">parentalNapaVsDMSODEG</span><span class="op">[</span><span class="va">retained</span>, <span class="op">]</span></span>
<span></span>
<span><span class="co">## Run one functional enrichment analysis using all significant genes</span></span>
<span><span class="co">## The species is homo sapiens ("hsapiens")</span></span>
<span><span class="co">## The g:SCS multiple testing correction method (Raudvere U et al 2019)</span></span>
<span><span class="co">## The WikiPathways database is used</span></span>
<span><span class="co">## Only the significant results are retained (significant=TRUE)</span></span>
<span><span class="co">## The evidence codes are included in the results (evcodes=TRUE)</span></span>
<span><span class="co">## A custom background included the tested genes is used</span></span>
<span><span class="va">gostres</span> <span class="op">&lt;-</span> <span class="fu">gprofiler2</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/gprofiler2/man/gost.html" class="external-link">gost</a></span><span class="op">(</span></span>
<span>                query<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>parental_napa_vs_DMSO<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">signRes</span><span class="op">$</span><span class="va">EnsemblID</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                organism<span class="op">=</span><span class="st">"hsapiens"</span>,</span>
<span>                correction_method<span class="op">=</span><span class="st">"g_SCS"</span>,</span>
<span>                sources<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"WP"</span><span class="op">)</span>,</span>
<span>                significant<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>                evcodes<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>                custom_bg<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">parentalNapaVsDMSODEG</span><span class="op">$</span><span class="va">EnsemblID</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><br></p>
<p>The gost() function returns an named list of 2 entries:</p>
<ul>
<li>The <strong>result</strong> entry contains the enrichment results.</li>
<li>The <strong>meta</strong> entry contains the metadata information.</li>
</ul>
<p><br></p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## The 'gostres' object is a list of 2 entries</span></span>
<span><span class="co">## The 'result' entry contains the enrichment results</span></span>
<span><span class="co">## The 'meta' entry contains the metadata information</span></span>
<span></span>
<span><span class="co">## Some columns of interest in the results</span></span>
<span><span class="va">gostres</span><span class="op">$</span><span class="va">result</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"query"</span>, <span class="st">"p_value"</span>, <span class="st">"term_size"</span>, </span>
<span>                    <span class="st">"query_size"</span>, <span class="st">"intersection_size"</span>, <span class="st">"term_id"</span><span class="op">)</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">##                   query      p_value term_size query_size intersection_size</span></span>
<span><span class="co">## 1 parental_napa_vs_DMSO 2.905549e-08        25        157                 7</span></span>
<span><span class="co">## 2 parental_napa_vs_DMSO 9.764963e-07        23        157                 6</span></span>
<span><span class="co">## 3 parental_napa_vs_DMSO 9.863615e-07       391        157                16</span></span>
<span><span class="co">## 4 parental_napa_vs_DMSO 2.065842e-06        13        157                 5</span></span>
<span><span class="co">##     term_id</span></span>
<span><span class="co">## 1 WP:WP3613</span></span>
<span><span class="co">## 2 WP:WP4925</span></span>
<span><span class="co">## 3 WP:WP3888</span></span>
<span><span class="co">## 4 WP:WP4211</span></span></code></pre>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## The term names can be longer than the one shown</span></span>
<span><span class="va">gostres</span><span class="op">$</span><span class="va">result</span><span class="op">[</span><span class="fl">19</span><span class="op">:</span><span class="fl">22</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"term_id"</span>, <span class="st">"source"</span>, <span class="st">"term_name"</span><span class="op">)</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">##      term_id source</span></span>
<span><span class="co">## 19 WP:WP2877     WP</span></span>
<span><span class="co">## 20 WP:WP5373     WP</span></span>
<span><span class="co">## 21  WP:WP516     WP</span></span>
<span><span class="co">## 22 WP:WP1742     WP</span></span>
<span><span class="co">##                                                                     term_name</span></span>
<span><span class="co">## 19                                                 Vitamin D receptor pathway</span></span>
<span><span class="co">## 20 Role of hypoxia angiogenesis and FGF pathway in OA chondrocyte hypertrophy</span></span>
<span><span class="co">## 21                                                          Hypertrophy model</span></span>
<span><span class="co">## 22                                                               TP53 network</span></span></code></pre>
<p><br></p>
</div>
<div class="section level3" number="6.2">
<h3 id="start-cytoscape">
<span class="header-section-number">6.2</span> Start Cytoscape<a class="anchor" aria-label="anchor" href="#start-cytoscape"></a>
</h3>
<p><a href="https://cytoscape.org/" class="external-link">Cytoscape</a> is an open source software for
visualizing networks. It enables network integration with any type of attribute
data. The Cytoscape software
is available at the <a href="https://cytoscape.org/" class="external-link">Cytoscape website</a>.</p>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:cytoscapeLogo01"></span>
<img src="cy3sticker.png" alt="Cytoscape software logo." width="55%"><p class="caption">
Figure 5: <span class="caption-title">Cytoscape software logo</span><br></p>
</div>
<p>The Cytoscape network generated by
<em><a href="https://github.com/adeschen/enrichViewNet" class="external-link">enrichViewNet</a></em>
will be automatically loaded into the <a href="https://cytoscape.org/" class="external-link">Cytoscape</a>
software when the application is running.</p>
<p>If the application is not running, a CX JSON file will be created. The file
can then be loaded manually into the <a href="https://cytoscape.org/" class="external-link">Cytoscape</a>
software.</p>
<p><br></p>
</div>
<div class="section level3" number="6.3">
<h3 id="create-a-gene-term-network">
<span class="header-section-number">6.3</span> Create a gene-term network<a class="anchor" aria-label="anchor" href="#create-a-gene-term-network"></a>
</h3>
<p>The gene-term network can be created with the <em>createNetwork()</em> function. If
<em>Cytoscape</em> is opened, the network should automatically be loaded in the
application. Otherwise, a CX JSON file is created. The CX JSON can be manually
be opened in <em>Cytoscape</em>.</p>
<p>The following figure shows what the gene-term network looks like
in <em>Cytoscape</em>. As there are numerous significant Reactome terms, the network
is a bit hectic.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Load saved enrichment results between parental Napa vs DMSO</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"parentalNapaVsDMSOEnrichment"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Create network for REACTOME significant terms</span></span>
<span><span class="co">## The 'removeRoot=TRUE' parameter removes the root term from the network</span></span>
<span><span class="co">## The network will either by created in Cytoscape (if the application is open)</span></span>
<span><span class="co">## or a CX file will be created in the temporary directory</span></span>
<span><span class="fu"><a href="../reference/createNetwork.html">createNetwork</a></span><span class="op">(</span>gostObject<span class="op">=</span><span class="va">parentalNapaVsDMSOEnrichment</span>,  source<span class="op">=</span><span class="st">"REAC"</span>, </span>
<span>        removeRoot<span class="op">=</span><span class="cn">TRUE</span>, title<span class="op">=</span><span class="st">"REACTOME_All"</span>, </span>
<span>        collection<span class="op">=</span><span class="st">"parental_napa_vs_DMSO"</span>, </span>
<span>        fileName<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>, <span class="st">"parentalNapaVsDMSOEnrichment.cx"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] TRUE</span></span></code></pre>
<p><br></p>
<p>This is an example of the Reactome network in <em>Cytoscape</em>.</p>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:networkInCytoscape"></span>
<img src="cytoscape_reactome_all_parental_napa_vs_DMSO.png" alt="All reactome terms in a gene-term network loaded in Cytoscape." width="110%"><p class="caption">
Figure 6: <span class="caption-title">All reactome terms in a gene-term network loaded in Cytoscape</span><br></p>
</div>
<p><br></p>
<p>To address this situation, an updated gene-term network containing only
Reactome terms of interest is created.</p>
<p><br></p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Load saved enrichment results between parental Napa vs DMSO</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"parentalNapaVsDMSOEnrichment"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## List of terms of interest</span></span>
<span><span class="va">reactomeSelected</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"REAC:R-HSA-9031628"</span>, <span class="st">"REAC:R-HSA-198725"</span>, </span>
<span>                        <span class="st">"REAC:R-HSA-9614085"</span>, <span class="st">"REAC:R-HSA-9617828"</span>,</span>
<span>                        <span class="st">"REAC:R-HSA-9614657"</span>, <span class="st">"REAC:R-HSA-73857"</span>,</span>
<span>                        <span class="st">"REAC:R-HSA-74160"</span>, <span class="st">"REAC:R-HSA-381340"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## All enrichment results</span></span>
<span><span class="va">results</span> <span class="op">&lt;-</span> <span class="va">parentalNapaVsDMSOEnrichment</span><span class="op">$</span><span class="va">result</span></span>
<span></span>
<span><span class="co">## Retain selected results</span></span>
<span><span class="va">selectedRes</span> <span class="op">&lt;-</span> <span class="va">results</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">results</span><span class="op">$</span><span class="va">term_id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">reactomeSelected</span><span class="op">)</span>, <span class="op">]</span></span>
<span></span>
<span><span class="co">## Print the first selected terms</span></span>
<span><span class="va">selectedRes</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"term_name"</span><span class="op">)</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "NGF-stimulated transcription"                                 </span></span>
<span><span class="co">## [2] "Nuclear Events (kinase and transcription factor activation)"  </span></span>
<span><span class="co">## [3] "FOXO-mediated transcription"                                  </span></span>
<span><span class="co">## [4] "FOXO-mediated transcription of cell death genes"              </span></span>
<span><span class="co">## [5] "RNA Polymerase II Transcription"                              </span></span>
<span><span class="co">## [6] "Gene expression (Transcription)"                              </span></span>
<span><span class="co">## [7] "Transcriptional regulation of white adipocyte differentiation"</span></span>
<span><span class="co">## [8] "FOXO-mediated transcription of cell cycle genes"</span></span></code></pre>
<p><br></p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Create network for REACTOME selected terms</span></span>
<span><span class="co">## The 'source="TERM_ID"' parameter enable to specify a personalized </span></span>
<span><span class="co">## list of terms of interest</span></span>
<span><span class="co">## The network will either by created in Cytoscape (if the application is open)</span></span>
<span><span class="co">## or a CX file will be created in the temporary directory</span></span>
<span><span class="fu"><a href="../reference/createNetwork.html">createNetwork</a></span><span class="op">(</span>gostObject<span class="op">=</span><span class="va">parentalNapaVsDMSOEnrichment</span>,  source<span class="op">=</span><span class="st">"TERM_ID"</span>, </span>
<span>        termIDs<span class="op">=</span><span class="va">selectedRes</span><span class="op">$</span><span class="va">term_id</span>, title<span class="op">=</span><span class="st">"REACTOME_Selected"</span>, </span>
<span>        collection<span class="op">=</span><span class="st">"parental_napa_vs_DMSO"</span>,</span>
<span>        fileName<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>, <span class="st">"parentalNapaVsDMSO_REACTOME.cx"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] TRUE</span></span></code></pre>
<p><br></p>
<p>The updated Reactome network in <em>Cytoscape</em>.</p>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:networkInCytoscapeSelected"></span>
<img src="cytoscape_with_selected_REACTOME_v01.png" alt="Selected Reactome terms in a gene-term network loaded in Cytoscape." width="110%"><p class="caption">
Figure 7: <span class="caption-title">Selected Reactome terms in a gene-term network loaded in Cytoscape</span><br></p>
</div>
<p><br></p>
<p>In <em>Cytoscape</em>, the appearance of a network is easily customized. As example,
default color and shape for all nodes can be modified. For this example,
the nodes have been moved to clarify their relation with the Reactome terms.</p>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:networkFinalReactome"></span>
<img src="REACTOME_Selected.jpeg" alt="Final Reactome network after customization inside Cytoscape." width="100%"><p class="caption">
Figure 8: <span class="caption-title">Final Reactome network after customization inside Cytoscape</span><br></p>
</div>
<p><br></p>
<p>The final Reactome network, after customization inside Cytoscape, shows that
multiple transcription enriched terms (<em>FOXO-mediated transcription</em>,
<em>FOXO-mediated transcription of cell cycle genes</em>,
<em>transcription regulation of white adipocyte differentiation</em>,
<em>RNA polymerase II transcription</em> and
<em>NGF-stimulated transcription</em> terms) are linked through enriched genes.</p>
<p><br><br></p>
</div>
</div>
<div class="section level2" number="7">
<h2 id="transforming-enrichment-results-into-an-enrichment-map">
<span class="header-section-number">7</span> Transforming enrichment results into an enrichment map<a class="anchor" aria-label="anchor" href="#transforming-enrichment-results-into-an-enrichment-map"></a>
</h2>
<p>The following workflow gives an overview of the steps associated to the
creation of an enrichment map.</p>
<p>The key steps for the workflow are:</p>
<div class="horizontal-scroll"><table class="table">
<colgroup>
<col width="43%">
<col width="56%">
</colgroup>
<thead><tr class="header">
<th>Step</th>
<th>Function</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>Run an enrichment analysis</td>
<td><code><a href="https://rdrr.io/pkg/gprofiler2/man/gost.html" class="external-link">gprofiler2::gost()</a></code></td>
</tr>
<tr class="even">
<td>Create an enrichment map</td>
<td><code><a href="../reference/createEnrichMap.html">createEnrichMap()</a></code></td>
</tr>
</tbody>
</table></div>
<p>The <code>package::function()</code> notation is used for functions from other packages.</p>
<p><br></p>
<div class="section level3" number="7.1">
<h3 id="run-an-enrichment-analysis-1">
<span class="header-section-number">7.1</span> Run an enrichment analysis<a class="anchor" aria-label="anchor" href="#run-an-enrichment-analysis-1"></a>
</h3>
<p>The first step consists in running an enrichment analysis with
<em><a href="https://CRAN.R-project.org/package=gprofiler2" class="external-link">gprofiler2</a></em> package. The output of the <em>gprofiler2::gost()</em>
is a <em>list</em> and should be saved.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Required library</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">gprofiler2</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## The dataset of differentially expressed genes done between </span></span>
<span><span class="co">## napabucasin treated and DMSO control parental (Froeling et al 2019)</span></span>
<span><span class="co">## All genes tested are present</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"parentalNapaVsDMSODEG"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Retain significant results </span></span>
<span><span class="co">## (absolute fold change superior to 1 and adjusted p-value inferior to 0.05)</span></span>
<span><span class="va">retained</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="va">parentalNapaVsDMSODEG</span><span class="op">$</span><span class="va">log2FoldChange</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">1</span> <span class="op">&amp;</span> </span>
<span>                        <span class="va">parentalNapaVsDMSODEG</span><span class="op">$</span><span class="va">padj</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span></span>
<span><span class="va">signRes</span> <span class="op">&lt;-</span> <span class="va">parentalNapaVsDMSODEG</span><span class="op">[</span><span class="va">retained</span>, <span class="op">]</span></span>
<span></span>
<span><span class="co">## Run one functional enrichment analysis using all significant genes</span></span>
<span><span class="co">## The species is homo sapiens ("hsapiens")</span></span>
<span><span class="co">## The g:SCS multiple testing correction method (Raudvere U et al 2019)</span></span>
<span><span class="co">## The WikiPathways database is used</span></span>
<span><span class="co">## Only the significant results are retained (significant=TRUE)</span></span>
<span><span class="co">## The evidence codes are included in the results (evcodes=TRUE)</span></span>
<span><span class="co">## A custom background included the tested genes is used</span></span>
<span><span class="va">gostres</span> <span class="op">&lt;-</span> <span class="fu">gprofiler2</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/gprofiler2/man/gost.html" class="external-link">gost</a></span><span class="op">(</span></span>
<span>                query<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>parental_napa_vs_DMSO<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">signRes</span><span class="op">$</span><span class="va">EnsemblID</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                organism<span class="op">=</span><span class="st">"hsapiens"</span>,</span>
<span>                correction_method<span class="op">=</span><span class="st">"g_SCS"</span>,</span>
<span>                sources<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"WP"</span><span class="op">)</span>,</span>
<span>                significant<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>                evcodes<span class="op">=</span><span class="cn">TRUE</span>,</span>
<span>                custom_bg<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">parentalNapaVsDMSODEG</span><span class="op">$</span><span class="va">EnsemblID</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><br></p>
<p>The <em>gost()</em> function returns an named list of 2 entries:</p>
<ul>
<li>The <strong>result</strong> entry contains the enrichment results.</li>
<li>The <strong>meta</strong> entry contains the metadata information.</li>
</ul>
<p><br></p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## The 'gostres' object is a list of 2 entries</span></span>
<span><span class="co">## The 'result' entry contains the enrichment results</span></span>
<span><span class="co">## The 'meta' entry contains the metadata information</span></span>
<span></span>
<span><span class="co">## Some columns of interest in the results</span></span>
<span><span class="va">gostres</span><span class="op">$</span><span class="va">result</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"query"</span>, <span class="st">"p_value"</span>, <span class="st">"term_size"</span>, </span>
<span>                    <span class="st">"query_size"</span>, <span class="st">"intersection_size"</span>, <span class="st">"term_id"</span><span class="op">)</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">##                   query      p_value term_size query_size intersection_size</span></span>
<span><span class="co">## 1 parental_napa_vs_DMSO 2.905549e-08        25        157                 7</span></span>
<span><span class="co">## 2 parental_napa_vs_DMSO 9.764963e-07        23        157                 6</span></span>
<span><span class="co">## 3 parental_napa_vs_DMSO 9.863615e-07       391        157                16</span></span>
<span><span class="co">## 4 parental_napa_vs_DMSO 2.065842e-06        13        157                 5</span></span>
<span><span class="co">##     term_id</span></span>
<span><span class="co">## 1 WP:WP3613</span></span>
<span><span class="co">## 2 WP:WP4925</span></span>
<span><span class="co">## 3 WP:WP3888</span></span>
<span><span class="co">## 4 WP:WP4211</span></span></code></pre>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## The term names can be longer than the one shown</span></span>
<span><span class="va">gostres</span><span class="op">$</span><span class="va">result</span><span class="op">[</span><span class="fl">19</span><span class="op">:</span><span class="fl">22</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"term_id"</span>, <span class="st">"source"</span>, <span class="st">"term_name"</span><span class="op">)</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">##      term_id source</span></span>
<span><span class="co">## 19 WP:WP2877     WP</span></span>
<span><span class="co">## 20 WP:WP5373     WP</span></span>
<span><span class="co">## 21  WP:WP516     WP</span></span>
<span><span class="co">## 22 WP:WP1742     WP</span></span>
<span><span class="co">##                                                                     term_name</span></span>
<span><span class="co">## 19                                                 Vitamin D receptor pathway</span></span>
<span><span class="co">## 20 Role of hypoxia angiogenesis and FGF pathway in OA chondrocyte hypertrophy</span></span>
<span><span class="co">## 21                                                          Hypertrophy model</span></span>
<span><span class="co">## 22                                                               TP53 network</span></span></code></pre>
<p><br></p>
</div>
<div class="section level3" number="7.2">
<h3 id="create-an-enrichment-map">
<span class="header-section-number">7.2</span> Create an enrichment map<a class="anchor" aria-label="anchor" href="#create-an-enrichment-map"></a>
</h3>
<p>The enrichment map can be created with the <em>createEnrichMap()</em> function. The
function generates a <em>ggplot</em> object.</p>
<p>In an enrichment map, terms with overlapping significant genes tend to
cluster together. The Jaccard correlation coefficient is used as a metric of
similarity. Terms with high similarity (similarity metric &gt; 0.2) are linked
together by edges. The edges are shorter and thicker when the similarity
metric is high.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Load saved enrichment results between parental Napa vs DMSO</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">parentalNapaVsDMSOEnrichment</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Set seed to ensure reproducible results</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">121</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Create network for all Kegg terms</span></span>
<span><span class="fu"><a href="../reference/createEnrichMap.html">createEnrichMap</a></span><span class="op">(</span>gostObject<span class="op">=</span><span class="va">parentalNapaVsDMSOEnrichment</span>, </span>
<span>                    query<span class="op">=</span><span class="st">"parental_napa_vs_DMSO"</span>, </span>
<span>                    source<span class="op">=</span><span class="st">"KEGG"</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:runCreateEmap01"></span>
<img src="enrichViewNet_files/figure-html/runCreateEmap01-1.png" alt="A Kegg enrichment map where terms with overlapping significant genes cluster together." width="100%"><p class="caption">
Figure 9: <span class="caption-title">A Kegg enrichment map where terms with overlapping significant genes cluster together</span><br></p>
</div>
<p><br></p>
<p>The Kegg enrichment map shows that the <em>MAPK signaling pathway</em> term is
highly influential in the network. In addition, the
<em>Transcriptional misregulation in cancer</em> term is the only isolated node.</p>
<p><br></p>
<div class="section level4" number="7.2.1">
<h4 id="using-list-of-term-ids">
<span class="header-section-number">7.2.1</span> Using list of term IDs<a class="anchor" aria-label="anchor" href="#using-list-of-term-ids"></a>
</h4>
<p>Instead of using a specific source, the enrichment map can be generated from a
personalized array of term IDs. To do so, the <em>source</em> parameter must be set
to <em>“TERM_ID”</em> and an array of term IDs must be passed to the <em>termIDs</em>
parameter. The terms can be pick-up from different sources. If more than one
terms have the same description, the term ID will be added at the end of the
description for clarity.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Load saved enrichment results between parental Napa vs DMSO</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">parentalNapaVsDMSOEnrichment</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## The term IDs must correspond to the IDs present in the "term_id" column</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">parentalNapaVsDMSOEnrichment</span><span class="op">$</span><span class="va">result</span><span class="op">[</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"query"</span>, <span class="st">"term_id"</span>, <span class="st">"term_name"</span><span class="op">)</span><span class="op">]</span>, </span>
<span>     n<span class="op">=</span><span class="fl">3</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                   query    term_id                                 term_name</span></span>
<span><span class="co">## 1 parental_napa_vs_DMSO GO:0048523   negative regulation of cellular process</span></span>
<span><span class="co">## 2 parental_napa_vs_DMSO GO:0008219                                cell death</span></span>
<span><span class="co">## 3 parental_napa_vs_DMSO GO:0048519 negative regulation of biological process</span></span></code></pre>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## List of selected terms from different sources</span></span>
<span><span class="va">termID</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"KEGG:04115"</span>, <span class="st">"WP:WP4963"</span>, <span class="st">"KEGG:04010"</span>, </span>
<span>                <span class="st">"REAC:R-HSA-5675221"</span>, <span class="st">"REAC:R-HSA-112409"</span>, <span class="st">"WP:WP382"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Set seed to ensure reproducible results</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">222</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Create network for all selected terms</span></span>
<span><span class="fu"><a href="../reference/createEnrichMap.html">createEnrichMap</a></span><span class="op">(</span>gostObject<span class="op">=</span><span class="va">parentalNapaVsDMSOEnrichment</span>, </span>
<span>                    query<span class="op">=</span><span class="st">"parental_napa_vs_DMSO"</span>, </span>
<span>                    source<span class="op">=</span><span class="st">"TERM_ID"</span>, termIDs<span class="op">=</span><span class="va">termID</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:runCreateEmapTerms"></span>
<img src="enrichViewNet_files/figure-html/runCreateEmapTerms-1.png" alt="An enrichment map showing only the user selected terms." width="100%"><p class="caption">
Figure 10: <span class="caption-title">An enrichment map showing only the user selected terms</span><br></p>
</div>
<p><br></p>
<p>As the description “MAPK signaling pathway” is present twice, the ID of each
term has been added to the end of the description. Hence, the MAPK pathway
from WikiPathway can be distinguished from the Kegg pathway.</p>
<p><br></p>
</div>
<div class="section level4" number="7.2.2">
<h4 id="effect-of-seed-value">
<span class="header-section-number">7.2.2</span> Effect of <em>seed</em> value<a class="anchor" aria-label="anchor" href="#effect-of-seed-value"></a>
</h4>
<p>The layering of the nodes is not always optimal. As the layering is affected by
the <em>seed</em> value, you might want to test few <em>seed</em> values, using the
<em>set.seed()</em> function, before selecting the final graph.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Set seed to ensure reproducible results</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">91</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Create network for all Kegg terms</span></span>
<span><span class="fu"><a href="../reference/createEnrichMap.html">createEnrichMap</a></span><span class="op">(</span>gostObject<span class="op">=</span><span class="va">parentalNapaVsDMSOEnrichment</span>, </span>
<span>                    query<span class="op">=</span><span class="st">"parental_napa_vs_DMSO"</span>, </span>
<span>                    source<span class="op">=</span><span class="st">"KEGG"</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:runCreateEmap02"></span>
<img src="enrichViewNet_files/figure-html/runCreateEmap02-1.png" alt="An enrichment map with a different seed." width="100%"><p class="caption">
Figure 11: <span class="caption-title">An enrichment map with a different seed</span><br></p>
</div>
<p><br></p>
</div>
<div class="section level4" number="7.2.3">
<h4 id="enrichment-map-customization">
<span class="header-section-number">7.2.3</span> Enrichment map customization<a class="anchor" aria-label="anchor" href="#enrichment-map-customization"></a>
</h4>
<p>The output of the <em>createEnrichMap()</em> function is a <em>ggplot</em> object.
This means that the graph can be personalized. For example, the default
colors can be changed:</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## The ggplot2 library is required</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Set seed to ensure reproducible results</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">91</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Create network for all Kegg terms</span></span>
<span><span class="va">graphKegg</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/createEnrichMap.html">createEnrichMap</a></span><span class="op">(</span>gostObject<span class="op">=</span><span class="va">parentalNapaVsDMSOEnrichment</span>, </span>
<span>                    query<span class="op">=</span><span class="st">"parental_napa_vs_DMSO"</span>, </span>
<span>                    source<span class="op">=</span><span class="st">"KEGG"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Nodes with lowest p-values will be in orange and highest p-values in black</span></span>
<span><span class="co">## The title of the legend is also modified</span></span>
<span><span class="va">graphKegg</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_gradient.html" class="external-link">scale_fill_gradient</a></span><span class="op">(</span>name<span class="op">=</span><span class="st">"P-value adjusted"</span>, low<span class="op">=</span><span class="st">"orange"</span>, </span>
<span>                                    high<span class="op">=</span><span class="st">"black"</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:runCreateEmap03"></span>
<img src="enrichViewNet_files/figure-html/runCreateEmap03-1.png" alt="An enrichment map with personalized colors." width="100%"><p class="caption">
Figure 12: <span class="caption-title">An enrichment map with personalized colors</span><br></p>
</div>
<p><br><br></p>
</div>
</div>
</div>
<div class="section level2" number="8">
<h2 id="transforming-enrichment-results-into-an-enrichment-map-with-groups-from-different-enrichment-analyses">
<span class="header-section-number">8</span> Transforming enrichment results into an enrichment map with groups from different enrichment analyses<a class="anchor" aria-label="anchor" href="#transforming-enrichment-results-into-an-enrichment-map-with-groups-from-different-enrichment-analyses"></a>
</h2>
<p>The following workflow gives an overview of the steps associated to the
creation of an enrichment map with groups. Groups can be created from the
same enrichment analysis or can come from different enrichment analyses. This
section is specifically for enrichment maps created using multiple enrichment
analyses.</p>
<p>The key steps for the workflow are:</p>
<div class="horizontal-scroll"><table class="table">
<colgroup>
<col width="60%">
<col width="40%">
</colgroup>
<thead><tr class="header">
<th>Step</th>
<th>Function</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>Run multiple enrichment analyses</td>
<td><code><a href="https://rdrr.io/pkg/gprofiler2/man/gost.html" class="external-link">gprofiler2::gost()</a></code></td>
</tr>
<tr class="even">
<td>Create an enrichment map with group</td>
<td><code><a href="../reference/createEnrichMapMultiBasic.html">createEnrichMapMultiBasic()</a></code></td>
</tr>
</tbody>
</table></div>
<p>The <code>package::function()</code> notation is used for functions from other packages.</p>
<p><br></p>
<p>The first step has been presented in the previous section.</p>
<p><br></p>
<div class="section level3" number="8.1">
<h3 id="create-an-enrichment-map-using-multiple-enrichment-analyses">
<span class="header-section-number">8.1</span> Create an enrichment map using multiple enrichment analyses<a class="anchor" aria-label="anchor" href="#create-an-enrichment-map-using-multiple-enrichment-analyses"></a>
</h3>
<p>An enrichment map can show enrichment results from multiple enrichment
analyses. A different color is used for each analysis. This enables to
highlight the enriched terms shared by multiples analyses as well as
analysis specific network of terms.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Set seed to ensure reproducible results</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">2121</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## The dataset of functional enriched terms for two experiments:</span></span>
<span><span class="co">## napabucasin treated and DMSO control parental and</span></span>
<span><span class="co">## napabucasin treated and DMSO control expressing Rosa26 control vector</span></span>
<span><span class="co">## (Froeling et al 2019)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"parentalNapaVsDMSOEnrichment"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"rosaNapaVsDMSOEnrichment"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## The gostObjectList is a list containing all </span></span>
<span><span class="co">## the functional enrichment objects</span></span>
<span><span class="va">gostObjectList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">parentalNapaVsDMSOEnrichment</span>, </span>
<span>    <span class="va">rosaNapaVsDMSOEnrichment</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## The queryList is a list of query names retained for each of the enrichment </span></span>
<span><span class="co">## object (same order). Beware that a enrichment object can contain more than </span></span>
<span><span class="co">## one query.</span></span>
<span><span class="va">query_01</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">parentalNapaVsDMSOEnrichment</span><span class="op">$</span><span class="va">result</span><span class="op">$</span><span class="va">query</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span></span>
<span><span class="va">query_02</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">rosaNapaVsDMSOEnrichment</span><span class="op">$</span><span class="va">result</span><span class="op">$</span><span class="va">query</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span></span>
<span><span class="va">queryList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">query_01</span>, <span class="va">query_02</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Enrichment map where the groups are the KEGG results for the 2 different</span></span>
<span><span class="co">## experiments</span></span>
<span><span class="fu"><a href="../reference/createEnrichMapMultiBasic.html">createEnrichMapMultiBasic</a></span><span class="op">(</span>gostObjectList<span class="op">=</span><span class="va">gostObjectList</span>,</span>
<span>    queryList<span class="op">=</span><span class="va">queryList</span>, source<span class="op">=</span><span class="st">"KEGG"</span>, removeRoot<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:emapMulti01"></span>
<img src="enrichViewNet_files/figure-html/emapMulti01-1.png" alt="An enrichment map containing Kegg enrichment results for 2 different experiments." width="100%"><p class="caption">
Figure 13: <span class="caption-title">An enrichment map containing Kegg enrichment results for 2 different experiments</span><br></p>
</div>
<p><br></p>
<p>There are 4 KEGG enrichment terms that are shared by the 2 experiments.
In additions, the <em>Viral carinogenesis</em> term is the only term that is not
related to the other KEGG terms.</p>
<p><br></p>
</div>
<div class="section level3" number="8.2">
<h3 id="effect-of-seed-value-1">
<span class="header-section-number">8.2</span> Effect of <em>seed</em> value<a class="anchor" aria-label="anchor" href="#effect-of-seed-value-1"></a>
</h3>
<p>The layering of the nodes is not always optimal. As the layering is affected by
the <em>seed</em> value, you might want to test few <em>seed</em> values, using the
<em>set.seed()</em> function, before selecting the final graph.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Set seed to ensure reproducible results</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">5</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Enrichment map where the groups are the KEGG results for the 2 different</span></span>
<span><span class="co">## experiments</span></span>
<span><span class="fu"><a href="../reference/createEnrichMapMultiBasic.html">createEnrichMapMultiBasic</a></span><span class="op">(</span>gostObjectList<span class="op">=</span><span class="va">gostObjectList</span>,</span>
<span>    queryList<span class="op">=</span><span class="va">queryList</span>, source<span class="op">=</span><span class="st">"KEGG"</span>, removeRoot<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:emapMultiSeed"></span>
<img src="enrichViewNet_files/figure-html/emapMultiSeed-1.png" alt="An enrichment map using the same data fromt the previous one but with a different seed." width="100%"><p class="caption">
Figure 14: <span class="caption-title">An enrichment map using the same data fromt the previous one but with a different seed</span><br></p>
</div>
<p><br></p>
</div>
<div class="section level3" number="8.3">
<h3 id="enrichment-map-customization-1">
<span class="header-section-number">8.3</span> Enrichment map customization<a class="anchor" aria-label="anchor" href="#enrichment-map-customization-1"></a>
</h3>
<p>The output of the <em>createEnrichMapMultiBasic()</em> function is a <em>ggplot</em> object.
This means that the graph can be personalized. For example, the default
colors, legend title and legend font can be changed:</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Required library</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Set seed to ensure reproducible results</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">5</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Enrichment map where the groups are the KEGG results for the 2 different</span></span>
<span><span class="co">## experiments</span></span>
<span><span class="fu"><a href="../reference/createEnrichMapMultiBasic.html">createEnrichMapMultiBasic</a></span><span class="op">(</span>gostObjectList<span class="op">=</span><span class="va">gostObjectList</span>,</span>
<span>    queryList<span class="op">=</span><span class="va">queryList</span>, source<span class="op">=</span><span class="st">"KEGG"</span>, removeRoot<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_fill_manual</a></span><span class="op">(</span>name<span class="op">=</span><span class="st">"Groups"</span>,</span>
<span>                breaks <span class="op">=</span> <span class="va">queryList</span>,</span>
<span>                values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"cyan4"</span>, <span class="st">"bisque3"</span><span class="op">)</span>,</span>
<span>                labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"parental"</span>, <span class="st">"rosa"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>legend.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>face<span class="op">=</span><span class="st">"bold"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<span style="display:block;" id="fig:emapMultiCustom"></span>
<img src="enrichViewNet_files/figure-html/emapMultiCustom-1.png" alt="An enrichment map with multiple groups and personalized colors." width="100%"><p class="caption">
Figure 15: <span class="caption-title">An enrichment map with multiple groups and personalized colors</span><br></p>
</div>
<p><br><br></p>
</div>
</div>
<div class="section level2" number="9">
<h2 id="transforming-enrichment-results-into-an-enrichment-map-with-groups-from-same-enrichment-analysis">
<span class="header-section-number">9</span> Transforming enrichment results into an enrichment map with groups from same enrichment analysis<a class="anchor" aria-label="anchor" href="#transforming-enrichment-results-into-an-enrichment-map-with-groups-from-same-enrichment-analysis"></a>
</h2>
<p>The following workflow gives an overview of the steps associated to the
creation of an enrichment map with groups. Groups can be created from the
same enrichment analysis or can come from different enrichment analyses. This
section is specifically for enrichment maps containing groups created using
one enrichment analysis.</p>
<p>The key steps for the workflow are:</p>
<div class="horizontal-scroll"><table class="table">
<colgroup>
<col width="60%">
<col width="40%">
</colgroup>
<thead><tr class="header">
<th>Step</th>
<th>Function</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>Run one enrichment analysis</td>
<td><code><a href="https://rdrr.io/pkg/gprofiler2/man/gost.html" class="external-link">gprofiler2::gost()</a></code></td>
</tr>
<tr class="even">
<td>Create an enrichment map with group</td>
<td><code>createEnrichMapMulti()</code></td>
</tr>
</tbody>
</table></div>
<p>The <code>package::function()</code> notation is used for functions from other packages.</p>
<p><br></p>
<p>The first step has been presented in the previous section.</p>
<p><br></p>
<div class="section level3" number="9.1">
<h3 id="create-an-enrichment-map-using-subsections-of-one-enrichment-analysis">
<span class="header-section-number">9.1</span> Create an enrichment map using subsections of one enrichment analysis<a class="anchor" aria-label="anchor" href="#create-an-enrichment-map-using-subsections-of-one-enrichment-analysis"></a>
</h3>
<p>An enrichment map can generate enrichment results from one enrichment
analysis with terms from different sources in different colors.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Set seed to ensure reproducible results</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">2121</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## The dataset of functional enriched terms for one experiment:</span></span>
<span><span class="co">## napabucasin treated and DMSO control expressing Rosa26 control vector</span></span>
<span><span class="co">## (Froeling et al 2019)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"rosaNapaVsDMSOEnrichment"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## The gostObjectList is a list containing all </span></span>
<span><span class="co">## the functional enrichment objects</span></span>
<span><span class="va">gostObjectList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">rosaNapaVsDMSOEnrichment</span>, </span>
<span>    <span class="va">rosaNapaVsDMSOEnrichment</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## The queryList is a list of query names retained for each of the enrichment </span></span>
<span><span class="co">## object (same order). Beware that a enrichment object can contain more than </span></span>
<span><span class="co">## one query.</span></span>
<span><span class="va">query_01</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">rosaNapaVsDMSOEnrichment</span><span class="op">$</span><span class="va">result</span><span class="op">$</span><span class="va">query</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span></span>
<span><span class="va">queryList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">query_01</span>, <span class="va">query_01</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Enrichment map where the groups are the KEGG results for the 2 different</span></span>
<span><span class="co">## experiments</span></span>
<span><span class="fu"><a href="../reference/createEnrichMapMultiBasic.html">createEnrichMapMultiBasic</a></span><span class="op">(</span>gostObjectList<span class="op">=</span><span class="va">gostObjectList</span>,</span>
<span>    queryList<span class="op">=</span><span class="va">queryList</span>, source<span class="op">=</span><span class="st">"KEGG"</span>, removeRoot<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<span style="display:block;" id="fig:emapMultiComplex01"></span>
<img src="enrichViewNet_files/figure-html/emapMultiComplex01-1.png" alt="An enrichment map containing TODO for one enrichment analysis." width="100%"><p class="caption">
Figure 16: <span class="caption-title">An enrichment map containing TODO for one enrichment analysis</span><br></p>
</div>
<p><br></p>
<p>TODO</p>
<p><br><br></p>
</div>
</div>
<div class="section level2" number="10">
<h2 id="acknowledgments">
<span class="header-section-number">10</span> Acknowledgments<a class="anchor" aria-label="anchor" href="#acknowledgments"></a>
</h2>
<p>The differentially expressed genes between napabucasin-treated cells (0.5 uM)
and DMSO as vehicle control are reprinted from
Clinical Cancer Research, 2019, 25 (23), 7162–7174,
Fieke E.M. Froeling, Manojit Mosur Swamynathan, Astrid Deschênes,
Iok In Christine Chio, Erin Brosnan, Melissa A. Yao, Priya Alagesan,
Matthew Lucito, Juying Li, An-Yun Chang, Lloyd C. Trotman, Pascal Belleau,
Youngkyu Park, Harry A. Rogoff, James D. Watson, David A. Tuveson,
Bioactivation of napabucasin triggers reactive oxygen species–mediated
cancer cell death, with permission from AACR.</p>
<p>Robert L. Faure is also supported by the National Sciences Engineering
Research Council of Canada (NSERCC): 155751-1501.</p>
<p><br><br></p>
</div>
<div class="section level2" number="11">
<h2 id="session-info">
<span class="header-section-number">11</span> Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<p>Here is the output of sessionInfo() on the system on which this document
was compiled:</p>
<pre><code><span><span class="co">## R Under development (unstable) (2024-04-03 r86327)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">## Running under: Ubuntu 22.04.4 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] ggplot2_3.5.0       gprofiler2_0.2.3    enrichViewNet_1.1.1</span></span>
<span><span class="co">## [4] knitr_1.46          BiocStyle_2.31.0   </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] RColorBrewer_1.1-3      jsonlite_1.8.8          magrittr_2.0.3         </span></span>
<span><span class="co">##   [4] farver_2.1.1            rmarkdown_2.26          fs_1.6.3               </span></span>
<span><span class="co">##   [7] zlibbioc_1.49.3         vctrs_0.6.5             memoise_2.0.1          </span></span>
<span><span class="co">##  [10] RCurl_1.98-1.14         ggtree_3.11.1           base64enc_0.1-3        </span></span>
<span><span class="co">##  [13] tinytex_0.50            htmltools_0.5.8.1       curl_5.2.1             </span></span>
<span><span class="co">##  [16] gridGraphics_0.5-1      strex_2.0.0             sass_0.4.9             </span></span>
<span><span class="co">##  [19] KernSmooth_2.23-22      bslib_0.7.0             htmlwidgets_1.6.4      </span></span>
<span><span class="co">##  [22] plyr_1.8.9              plotly_4.10.4           cachem_1.0.8           </span></span>
<span><span class="co">##  [25] uuid_1.2-0              igraph_2.0.3            lifecycle_1.0.4        </span></span>
<span><span class="co">##  [28] pkgconfig_2.0.3         Matrix_1.7-0            R6_2.5.1               </span></span>
<span><span class="co">##  [31] fastmap_1.1.1           GenomeInfoDbData_1.2.12 digest_0.6.35          </span></span>
<span><span class="co">##  [34] aplot_0.2.2             enrichplot_1.23.1       ggnewscale_0.4.10      </span></span>
<span><span class="co">##  [37] colorspace_2.1-0        patchwork_1.2.0         AnnotationDbi_1.65.2   </span></span>
<span><span class="co">##  [40] S4Vectors_0.41.6        RSQLite_2.3.6           base64url_1.4          </span></span>
<span><span class="co">##  [43] labeling_0.4.3          fansi_1.0.6             RJSONIO_1.3-1.9        </span></span>
<span><span class="co">##  [46] httr_1.4.7              polyclip_1.10-6         compiler_4.5.0         </span></span>
<span><span class="co">##  [49] bit64_4.0.5             withr_3.0.0             backports_1.4.1        </span></span>
<span><span class="co">##  [52] BiocParallel_1.37.1     viridis_0.6.5           RCy3_2.23.3            </span></span>
<span><span class="co">##  [55] DBI_1.2.2               highr_0.10              ggforce_0.4.2          </span></span>
<span><span class="co">##  [58] gplots_3.1.3.1          MASS_7.3-60.2           HDO.db_0.99.1          </span></span>
<span><span class="co">##  [61] gtools_3.9.5            caTools_1.18.2          tools_4.5.0            </span></span>
<span><span class="co">##  [64] scatterpie_0.2.2        ape_5.7-1               glue_1.7.0             </span></span>
<span><span class="co">##  [67] nlme_3.1-164            GOSemSim_2.29.1         shadowtext_0.1.3       </span></span>
<span><span class="co">##  [70] grid_4.5.0              checkmate_2.3.1         pbdZMQ_0.3-11          </span></span>
<span><span class="co">##  [73] reshape2_1.4.4          fgsea_1.29.0            generics_0.1.3         </span></span>
<span><span class="co">##  [76] gtable_0.3.4            tidyr_1.3.1             data.table_1.15.4      </span></span>
<span><span class="co">##  [79] tidygraph_1.3.1         utf8_1.2.4              XVector_0.43.1         </span></span>
<span><span class="co">##  [82] BiocGenerics_0.49.1     ggrepel_0.9.5           pillar_1.9.0           </span></span>
<span><span class="co">##  [85] stringr_1.5.1           yulab.utils_0.1.4       IRdisplay_1.1          </span></span>
<span><span class="co">##  [88] splines_4.5.0           dplyr_1.1.4             tweenr_2.0.3           </span></span>
<span><span class="co">##  [91] treeio_1.27.0           lattice_0.22-6          bit_4.0.5              </span></span>
<span><span class="co">##  [94] tidyselect_1.2.1        GO.db_3.19.0            Biostrings_2.71.5      </span></span>
<span><span class="co">##  [97] gridExtra_2.3           bookdown_0.38           IRanges_2.37.1         </span></span>
<span><span class="co">## [100] stats4_4.5.0            xfun_0.43               graphlayouts_1.1.1     </span></span>
<span><span class="co">## [103] Biobase_2.63.1          stringi_1.8.3           UCSC.utils_0.99.5      </span></span>
<span><span class="co">## [106] lazyeval_0.2.2          ggfun_0.1.4             yaml_2.3.8             </span></span>
<span><span class="co">## [109] evaluate_0.23           codetools_0.2-20        ggraph_2.2.1           </span></span>
<span><span class="co">## [112] tibble_3.2.1            qvalue_2.35.0           BiocManager_1.30.22    </span></span>
<span><span class="co">## [115] graph_1.81.0            ggplotify_0.1.2         cli_3.6.2              </span></span>
<span><span class="co">## [118] IRkernel_1.3.2          repr_1.1.7              munsell_0.5.1          </span></span>
<span><span class="co">## [121] jquerylib_0.1.4         Rcpp_1.0.12             GenomeInfoDb_1.39.13   </span></span>
<span><span class="co">## [124] png_0.1-8               XML_3.99-0.16.1         parallel_4.5.0         </span></span>
<span><span class="co">## [127] blob_1.2.4              DOSE_3.29.2             bitops_1.0-7           </span></span>
<span><span class="co">## [130] viridisLite_0.4.2       tidytree_0.4.6          scales_1.3.0           </span></span>
<span><span class="co">## [133] purrr_1.0.2             crayon_1.5.2            rlang_1.1.3            </span></span>
<span><span class="co">## [136] cowplot_1.1.3           fastmatch_1.1-4         KEGGREST_1.43.0</span></span></code></pre>
<p><br><br></p>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Dennis2003" class="csl-entry">
Dennis, Glynn, Brad T. Sherman, Douglas A. Hosack, Jun Yang, Wei Gao, H. Clifford Lane, and Richard A. Lempicki. 2003. <span>“<span class="nocase">DAVID: Database for Annotation, Visualization, and Integrated Discovery.</span>”</span> <em>Genome Biology</em> 4 (5). <a href="https://doi.org/10.1186/gb-2003-4-9-r60" class="external-link">https://doi.org/10.1186/gb-2003-4-9-r60</a>.
</div>
<div id="ref-Kolberg2020" class="csl-entry">
Kolberg, Liis, Uku Raudvere, Ivan Kuzmin, Jaak Vilo, and Hedi Peterson. 2020. <span>“<span class="nocase">gprofiler2 – an R package for gene list functional enrichment analysis and namespace conversion toolset g:Profiler</span>.”</span> <em>F1000Research</em> 9: 1–27. <a href="https://doi.org/10.12688/f1000research.24956.1" class="external-link">https://doi.org/10.12688/f1000research.24956.1</a>.
</div>
<div id="ref-Kuleshov2016" class="csl-entry">
Kuleshov, Maxim V., Matthew R. Jones, Andrew D. Rouillard, Nicolas F. Fernandez, Qiaonan Duan, Zichen Wang, Simon Koplev, et al. 2016. <span>“<span class="nocase">Enrichr: a comprehensive gene set enrichment analysis web server 2016 update</span>.”</span> <em>Nucleic Acids Research</em> 44 (W1): W90–97. <a href="https://doi.org/10.1093/nar/gkw377" class="external-link">https://doi.org/10.1093/nar/gkw377</a>.
</div>
<div id="ref-Maere2005" class="csl-entry">
Maere, Steven, Karel Heymans, and Martin Kuiper. 2005. <span>“<span class="nocase">BiNGO: a Cytoscape plugin to assess overrepresentation of Gene Ontology categories in Biological Networks</span>.”</span> <em>Bioinformatics</em> 21 (16): 3448–49. <a href="https://doi.org/10.1093/bioinformatics/bti551" class="external-link">https://doi.org/10.1093/bioinformatics/bti551</a>.
</div>
<div id="ref-Raudvere2019" class="csl-entry">
Raudvere, Uku, Liis Kolberg, Ivan Kuzmin, Tambet Arak, Priit Adler, Hedi Peterson, and Jaak Vilo. 2019. <span>“<span class="nocase">G:Profiler: A web server for functional enrichment analysis and conversions of gene lists (2019 update)</span>.”</span> <em>Nucleic Acids Research</em> 47 (W1): W191–98. <a href="https://doi.org/10.1093/nar/gkz369" class="external-link">https://doi.org/10.1093/nar/gkz369</a>.
</div>
<div id="ref-Scardoni2009" class="csl-entry">
Scardoni, Giovanni, Michele Petterlini, and Carlo Laudanna. 2009. <span>“<span class="nocase">Analyzing biological network parameters with CentiScaPe</span>.”</span> <em>Bioinformatics</em> 25 (21): 2857–59. <a href="https://doi.org/10.1093/bioinformatics/btp517" class="external-link">https://doi.org/10.1093/bioinformatics/btp517</a>.
</div>
<div id="ref-PaulShannon2003" class="csl-entry">
Shannon, Paul, Andrew Markiel, Owen Ozier, Nitin S Baliga, Jonathan T Wang, Daniel Ramage, Nada Amin, Benno Schwikowski, and Trey Ideker. 2003. <span>“<span class="nocase">Cytoscape: a software environment for integrated models of biomolecular interaction networks.</span>”</span> <em>Genome Research</em> 13 (11): 2498–2504. <a href="https://doi.org/10.1101/gr.1239303" class="external-link">https://doi.org/10.1101/gr.1239303</a>.
</div>
<div id="ref-Yu2022" class="csl-entry">
Yu, Guangchuang. 2022. <em>Enrichplot: Visualization of Functional Enrichment Result</em>. <a href="https://yulab-smu.top/biomedical-knowledge-mining-book/" class="external-link">https://yulab-smu.top/biomedical-knowledge-mining-book/</a>.
</div>
</div>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script><!-- tabsets --><script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script><!-- code folding --><script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": {
      styles: {
        ".MathJax_Display": {
           "text-align": "center",
           padding: "0px 150px 0px 65px",
           margin: "0px 0px 0.5em"
        },
        "@media screen and (max-width: 991px)": {
            ".MathJax_Display": {
               "text-align": "center",
               padding: "0 0 0 0"
            }
         }
      }
    }
  });
</script><script type="text/javascript">
$(document).ready(function ()  {
  
  // Map "enter" keypress to the same action as a cursor click
  function navigateLink(e) {
    if (e.key === "Enter") {
      $(this).trigger("click");
    }
  }

  var toc_items = document.querySelectorAll(".tocify-item");
  for (var i = 0; i < toc_items.length; i++) {
    // The link role tells screen readers this is for navigation
    toc_items.item(i).setAttribute("role", "link");
    // tabindex = 0 allows selection via keyboard tab presses
    toc_items.item(i).setAttribute("tabindex", "0");
    // Listen for "Enter" keypress when item is selected
    toc_items.item(i).addEventListener("keydown", navigateLink);
  }
});
</script>
</body>
</html>
