<!DOCTYPE html>
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="pandoc">
<meta http-equiv="X-UA-Compatible" content="IE=EDGE">
<meta name="author" content="Astrid Deschênes and Pascal Belleau">
<meta name="date" content="2021-08-19">
<title>From gprofiler2 to Cytoscape network</title>
<script src="gprofiler2cytoscape_files/header-attrs-2.10/header-attrs.js"></script><script src="gprofiler2cytoscape_files/jquery-1.11.3/jquery.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1">
<link href="gprofiler2cytoscape_files/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet">
<script src="gprofiler2cytoscape_files/bootstrap-3.3.5/js/bootstrap.min.js"></script><script src="gprofiler2cytoscape_files/bootstrap-3.3.5/shim/html5shiv.min.js"></script><script src="gprofiler2cytoscape_files/bootstrap-3.3.5/shim/respond.min.js"></script><style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
</style>
<script src="gprofiler2cytoscape_files/navigation-1.1/tabsets.js"></script><style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>
<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script><style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>
<link rel="stylesheet" href="/__w/_temp/Library/BiocStyle/resources/html/bioconductor.css" type="text/css">
<style type="text/css">
.main-container {
  max-width: 828px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>
<!-- tabsets --><style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>
<!-- code folding --><script>
function toggle_visibility(id1) {
  var e = document.getElementById(id1);
  e.style.display = ((e.style.display!="none") ? "none" : "block");
}
</script>
</head>
<body>


<div class="container-fluid main-container">




<div id="header">



<h1 class="hasAnchor">
<a href="#header" class="anchor"></a>From gprofiler2 to Cytoscape network</h1>
<p class="author-name">Astrid Deschênes and Pascal Belleau</p>
<h4 class="date">19 August 2021</h4>

</div>

<h1>Contents</h1>
<div id="TOC">
<ul>
<li><a href="#licensing"><span class="toc-section-number">1</span> Licensing</a></li>
<li><a href="#citing"><span class="toc-section-number">2</span> Citing</a></li>
<li><a href="#introduction"><span class="toc-section-number">3</span> Introduction</a></li>
<li><a href="#installation"><span class="toc-section-number">4</span> Installation</a></li>
<li><a href="#transforming-gprofiler2-enrichment-results-to-cytoscape-network"><span class="toc-section-number">5</span> Transforming gprofiler2 enrichment results to Cytoscape network</a></li>
<li><a href="#session-info"><span class="toc-section-number">6</span> Session info</a></li>
<li><a href="#references">References</a></li>
</ul>
</div>

<p><br><strong>Package</strong>: <em>gprofiler2cytoscape</em><br><strong>Authors</strong>: Astrid Deschênes [aut, cre] (<a href="https://orcid.org/0000-0001-7846-6749" class="uri">https://orcid.org/0000-0001-7846-6749</a>),
Pascal Belleau [aut] (<a href="https://orcid.org/0000-0002-0802-1071" class="uri">https://orcid.org/0000-0002-0802-1071</a>)<br><strong>Version</strong>: 0.0.3<br><strong>Compiled date</strong>: 2021-08-19<br><strong>License</strong>: Artistic-2.0<br></p>
<div id="licensing" class="section level1" number="1">
<h1 class="hasAnchor">
<a href="#licensing" class="anchor"></a><span class="header-section-number">1</span> Licensing</h1>
<p>The <em><a href="https://github.com/adeschen/gprofiler2cytoscape">gprofiler2cytoscape</a></em> package and the underlying
<em><a href="https://github.com/adeschen/gprofiler2cytoscape">gprofiler2cytoscape</a></em> code are distributed under the
Artistic license 2.0. You are free to use and redistribute this software.</p>
</div>
<div id="citing" class="section level1" number="2">
<h1 class="hasAnchor">
<a href="#citing" class="anchor"></a><span class="header-section-number">2</span> Citing</h1>
<p>If you use this package for a publication, we would ask you to cite the
following:</p>
<blockquote>
<p>Astrid Deschênes and Pascal Belleau (2021). gprofiler2cytoscape: From
gprofiler2 to Cytoscape network.
<a href="https://github.com/adeschen/gprofiler2cytoscape" class="uri">https://github.com/adeschen/gprofiler2cytoscape</a>,
<a href="https://adeschen.github.io/gprofiler2cytoscape/" class="uri">https://adeschen.github.io/gprofiler2cytoscape/</a>.</p>
</blockquote>
</div>
<div id="introduction" class="section level1" number="3">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a><span class="header-section-number">3</span> Introduction</h1>
<p>Biology studies that use high-throughput technologies, such as RNA sequencing
or microarray platform, usually end up generating large amount of data such
as gene lists. For instance, RNA sequencing is frequently uses to discover
genes that have differential expression between two biological states
(such as healthy versus diseased). Those studies can make use of statistical
tools that enable positioning those novel gene lists against previous
knowledge.</p>
<p>Functional enrichment analysis tools, such as <em>Enrichr</em> <span class="citation">(Kuleshov et al. 2016)</span> and
<em>DAVID</em> <span class="citation">(Dennis et al. 2003)</span>, specialize in positioning novel findings against
well curated data sources of biological processes and pathways. Most
specifically, those tools identify functionally coherent gene sets that are
statistically over- (or under-) represented in a gene list. The
traditional output of enrichment analysis tool is a table containing the
significant gene sets with their associated statistics. While those results
are extremely useful, their interpretation is not always easy. In that
respect, visual representation of the results as network could be handy.</p>
<p>Biological network models are visual representation of various biological
interacting elements which are based on mathematical graphs. In those
networks, the biological elements are generally represented by nodes while
the interactions and relationships are represented by edges. One of the
wildly used network tool in the quantitative biology community is the
open source software <em>Cytoscape</em>. In addition of biological data
visualization and network
analysis, <em>Cytoscape</em> can be expended through the use of
specialized plug-ins such as <em>BiNGO</em> that calculates over-represented GO terms
in a network <span class="citation">(Maere, Heymans, and Kuiper 2005)</span> or <em>CentiScaPe</em> that identifies relevant network
nodes <span class="citation">(Scardoni, Petterlini, and Laudanna 2009)</span>.</p>
<p>The <em>g:Profiler</em> enrichment analysis tool <span class="citation">(Raudvere et al. 2019)</span> is web based as
the particularity of being accompanied by the CRAN package
<em>gprofiler2</em> <span class="citation">(Kolberg et al. 2020)</span>. The <em>gprofiler2</em> package enables researchers to
incorporate functional enrichment analysis into automated analysis pipelines
written in R. This greatly facilitates research reproducibility.</p>
<p>The <em><a href="https://github.com/adeschen/gprofiler2cytoscape">gprofiler2cytoscape</a></em> package enables the
visualization of enrichment results generated by <em>gprofiler2</em> as
customisable <em>Cytoscape</em> network <span class="citation">(Shannon et al. 2003)</span>. In the biological
networks generated by <em><a href="https://github.com/adeschen/gprofiler2cytoscape">gprofiler2cytoscape</a></em>, both
gene datasets (GO terms/pathways/protein complexes) and genes associated
to the datasets are represented as nodes. While the edges connect each gene
to its dataset(s). Only genes present in the query used for the
enrichment analysis are shown.</p>
<div class="figure" style="text-align: left">
<span style="display:block;" id="fig:graphDemo01"></span>
<img src="demo01.jpeg" alt="Demo a biologocial network where GO terms and genes are presented as nodes while edges connect each gene to its dataset(s)." width="90%"><p class="caption">
Figure 1: <span class="caption-title">Demo a biologocial network where GO terms and genes are presented as nodes while edges connect each gene to its dataset(s)</span><br></p>
</div>
<p>The <strong><em>gprofiler2cytoscape</em></strong> package
offers the option to generate a network for only a portion of the
significant terms by selecting the source or by providing a
specific list of terms.</p>
<p>Once the network is created, the user can personalize the visual attributes
and integrate external information such as expression profiles, phenotypes
and other molecular states. The user can also peforms network analysis.</p>
</div>
<div id="installation" class="section level1" number="4">
<h1 class="hasAnchor">
<a href="#installation" class="anchor"></a><span class="header-section-number">4</span> Installation</h1>
<p>To install the latest version accessible on the
<em><a href="https://github.com/adeschen/gprofiler2cytoscape">gprofiler2cytoscape</a></em> Github Website, the
<em><a href="https://CRAN.R-project.org/package=devtools">devtools</a></em> package is required.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Load required package</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://devtools.r-lib.org/">devtools</a></span><span class="op">)</span>

<span class="co">## Install the latest version of gprofiler2cytoscape</span>
<span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">'adeschen/gprofiler2cytoscape'</span><span class="op">)</span></code></pre></div>
<p>It is also possible to install an official release. The list of available
releases is posted on the <a href="https://github.com/adeschen/gprofiler2cytoscape/releases" title="gprofiler2cytoscape Release Site">gprofiler2cytoscape Release Website</a>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Load required package</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://devtools.r-lib.org/">devtools</a></span><span class="op">)</span>

<span class="co">## Install the version v0.1 of gprofiler2cytoscape</span>
<span class="co">## using 'ref' parameter</span>
<span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">'adeschen/gprofiler2cytoscape'</span>, ref <span class="op">=</span> <span class="st">"v0.1"</span><span class="op">)</span></code></pre></div>
<p>TODO</p>
</div>
<div id="transforming-gprofiler2-enrichment-results-to-cytoscape-network" class="section level1" number="5">
<h1 class="hasAnchor">
<a href="#transforming-gprofiler2-enrichment-results-to-cytoscape-network" class="anchor"></a><span class="header-section-number">5</span> Transforming gprofiler2 enrichment results to Cytoscape network</h1>
<div class="figure" style="text-align: left">
<span style="display:block;" id="fig:graphListToGraph01"></span>
<img src="FromListToGraph_v01.jpg" alt="From an enrichment list to a Cytoscape network" width="100%"><p class="caption">
Figure 2: <span class="caption-title">From an enrichment list to a Cytoscape network</span><br></p>
</div>
</div>
<div id="session-info" class="section level1" number="6">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a><span class="header-section-number">6</span> Session info</h1>
<p>Here is the output of sessionInfo() on the system on which this document
was compiled:</p>
<pre><code>## R version 4.1.0 (2021-05-18)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 20.04.2 LTS
## 
## Matrix products: default
## BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] gprofiler2cytoscape_0.0.3 knitr_1.33               
## [3] BiocStyle_2.20.2         
## 
## loaded via a namespace (and not attached):
##  [1] httr_1.4.2          tidyr_1.1.3         jsonlite_1.7.2     
##  [4] viridisLite_0.4.0   R.utils_2.10.1      gprofiler2_0.2.0   
##  [7] BiocManager_1.30.16 highr_0.9           stats4_4.1.0       
## [10] dplR_1.7.2          base64url_1.4       yaml_2.2.1         
## [13] pillar_1.6.2        backports_1.2.1     lattice_0.20-44    
## [16] glue_1.4.2          uuid_0.1-4          digest_0.6.27      
## [19] RColorBrewer_1.1-2  uchardet_1.1.0      colorspace_2.0-2   
## [22] htmltools_0.5.1.1   Matrix_1.3-4        R.oo_1.24.0        
## [25] plyr_1.8.6          XML_3.99-0.7        pkgconfig_2.0.3    
## [28] bookdown_0.23       purrr_0.3.4         scales_1.1.1       
## [31] tibble_3.1.3        generics_0.1.0      ggplot2_3.3.5      
## [34] ellipsis_0.3.2      repr_1.1.3          BiocGenerics_0.38.0
## [37] lazyeval_0.2.2      RJSONIO_1.3-1.5     magrittr_2.0.1     
## [40] crayon_1.4.1        strex_1.4.2         evaluate_0.14      
## [43] R.methodsS3_1.8.1   fansi_0.5.0         MASS_7.3-54        
## [46] graph_1.70.0        tools_4.1.0         data.table_1.14.0  
## [49] lifecycle_1.0.0     matrixStats_0.60.0  stringr_1.4.0      
## [52] plotly_4.9.4.1      munsell_0.5.0       compiler_4.1.0     
## [55] signal_0.7-7        rlang_0.4.11        grid_4.1.0         
## [58] RCurl_1.98-1.4      pbdZMQ_0.3-5        IRkernel_1.2       
## [61] htmlwidgets_1.5.3   igraph_1.2.6        bitops_1.0-7       
## [64] base64enc_0.1-3     rmarkdown_2.10      gtable_0.3.0       
## [67] RCy3_2.12.4         R6_2.5.0            dplyr_1.0.7        
## [70] utf8_1.2.2          stringi_1.7.3       parallel_4.1.0     
## [73] IRdisplay_1.0       Rcpp_1.0.7          vctrs_0.3.8        
## [76] png_0.1-7           tidyselect_1.1.1    xfun_0.25</code></pre>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Dennis2003" class="csl-entry">
Dennis, Glynn, Brad T. Sherman, Douglas A. Hosack, Jun Yang, Wei Gao, H. Clifford Lane, and Richard A. Lempicki. 2003. <span>“<span class="nocase">DAVID: Database for Annotation, Visualization, and Integrated Discovery.</span>”</span> <em>Genome Biology</em> 4 (5). <a href="https://doi.org/10.1186/gb-2003-4-9-r60">https://doi.org/10.1186/gb-2003-4-9-r60</a>.
</div>
<div id="ref-Kolberg2020" class="csl-entry">
Kolberg, Liis, Uku Raudvere, Ivan Kuzmin, Jaak Vilo, and Hedi Peterson. 2020. <span>“<span class="nocase">gprofiler2 – an R package for gene list functional enrichment analysis and namespace conversion toolset g:Profiler</span>.”</span> <em>F1000Research</em> 9: 1–27. <a href="https://doi.org/10.12688/f1000research.24956.1">https://doi.org/10.12688/f1000research.24956.1</a>.
</div>
<div id="ref-Kuleshov2016" class="csl-entry">
Kuleshov, Maxim V., Matthew R. Jones, Andrew D. Rouillard, Nicolas F. Fernandez, Qiaonan Duan, Zichen Wang, Simon Koplev, et al. 2016. <span>“<span class="nocase">Enrichr: a comprehensive gene set enrichment analysis web server 2016 update</span>.”</span> <em>Nucleic Acids Research</em> 44 (W1): W90–97. <a href="https://doi.org/10.1093/nar/gkw377">https://doi.org/10.1093/nar/gkw377</a>.
</div>
<div id="ref-Maere2005" class="csl-entry">
Maere, Steven, Karel Heymans, and Martin Kuiper. 2005. <span>“<span class="nocase">BiNGO: a Cytoscape plugin to assess overrepresentation of Gene Ontology categories in Biological Networks</span>.”</span> <em>Bioinformatics</em> 21 (16): 3448–49. <a href="https://doi.org/10.1093/bioinformatics/bti551">https://doi.org/10.1093/bioinformatics/bti551</a>.
</div>
<div id="ref-Raudvere2019" class="csl-entry">
Raudvere, Uku, Liis Kolberg, Ivan Kuzmin, Tambet Arak, Priit Adler, Hedi Peterson, and Jaak Vilo. 2019. <span>“<span class="nocase">G:Profiler: A web server for functional enrichment analysis and conversions of gene lists (2019 update)</span>.”</span> <em>Nucleic Acids Research</em> 47 (W1): W191–98. <a href="https://doi.org/10.1093/nar/gkz369">https://doi.org/10.1093/nar/gkz369</a>.
</div>
<div id="ref-Scardoni2009" class="csl-entry">
Scardoni, Giovanni, Michele Petterlini, and Carlo Laudanna. 2009. <span>“<span class="nocase">Analyzing biological network parameters with CentiScaPe</span>.”</span> <em>Bioinformatics</em> 25 (21): 2857–59. <a href="https://doi.org/10.1093/bioinformatics/btp517">https://doi.org/10.1093/bioinformatics/btp517</a>.
</div>
<div id="ref-PaulShannon2003" class="csl-entry">
Shannon, Paul, Andrew Markiel, Owen Ozier, Nitin S Baliga, Jonathan T Wang, Daniel Ramage, Nada Amin, Benno Schwikowski, and Trey Ideker. 2003. <span>“<span class="nocase">Cytoscape: a software environment for integrated models of biomolecular interaction networks.</span>”</span> <em>Genome Research</em> 13 (11): 2498–2504. <a href="https://doi.org/10.1101/gr.1239303">https://doi.org/10.1101/gr.1239303</a>.
</div>
</div>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script><!-- tabsets --><script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script><!-- code folding --><script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": {
      styles: {
        ".MathJax_Display": {
           "text-align": "center",
           padding: "0px 150px 0px 65px",
           margin: "0px 0px 0.5em"
        },
        "@media screen and (max-width: 991px)": {
            ".MathJax_Display": {
               "text-align": "center",
               padding: "0 0 0 0"
            }
         }
      }
    }
  });
</script>
</body>
</html>
