<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>From gprofiler2 to Cytoscape network • gprofiler2cytoscape</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="From gprofiler2 to Cytoscape network">
<meta property="og:description" content="gprofiler2cytoscape">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gprofiler2cytoscape</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/gprofiler2cytoscape.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/adeschen/gprofiler2cytoscape/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="gprofiler2cytoscape_files/header-attrs-2.6/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>From gprofiler2 to Cytoscape network</h1>
                        <h4 class="author">Astrid Deschênes and Pascal Belleau</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/adeschen/gprofiler2cytoscape/blob/master/vignettes/gprofiler2cytoscape.Rmd"><code>vignettes/gprofiler2cytoscape.Rmd</code></a></small>
      <div class="hidden name"><code>gprofiler2cytoscape.Rmd</code></div>

    </div>

    
    
<p><br><strong>Package</strong>: <em>gprofiler2cytoscape</em><br><strong>Authors</strong>: Astrid Deschênes [aut, cre] (<a href="https://orcid.org/0000-0001-7846-6749" class="uri">https://orcid.org/0000-0001-7846-6749</a>), Pascal Belleau [aut] (<a href="https://orcid.org/0000-0002-0802-1071" class="uri">https://orcid.org/0000-0002-0802-1071</a>)<br><strong>Version</strong>: 0.0.1<br><strong>Compiled date</strong>: 2021-04-16<br><strong>License</strong>: Artistic-2.0<br></p>
<div id="licensing" class="section level1">
<h1 class="hasAnchor">
<a href="#licensing" class="anchor"></a>Licensing</h1>
<p>The <em><a href="https://github.com/adeschen/gprofiler2cytoscape">gprofiler2cytoscape</a></em> package and the underlying <em><a href="https://github.com/adeschen/gprofiler2cytoscape">gprofiler2cytoscape</a></em> code are distributed under the Artistic license 2.0. You are free to use and redistribute this software.</p>
</div>
<div id="citing" class="section level1">
<h1 class="hasAnchor">
<a href="#citing" class="anchor"></a>Citing</h1>
<p>If you use this package for a publication, we would ask you to cite the following:</p>
<blockquote>
<p>Astrid Deschênes and Pascal Belleau (2021). gprofiler2cytoscape: From gprofiler2 to Cytoscape network. <a href="https://github.com/adeschen/gprofiler2cytoscape" class="uri">https://github.com/adeschen/gprofiler2cytoscape</a>, <a href="https://adeschen.github.io/gprofiler2cytoscape/" class="uri">https://adeschen.github.io/gprofiler2cytoscape/</a>.</p>
</blockquote>
</div>
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>Biology studies that use high-throughput technologies, such as RNA sequencing or microarray platform, usually end up generating large amount of data such as gene lists. For instance, RNA sequencing is frequently uses to discover genes that have differential expression between two biological states (such as healthy versus diseased). Those studies can make use of statistical tools that enable positioning those novel gene lists against previous knowledge.</p>
<p>Functional enrichment analysis tools, such as <em>Enrichr</em> <span class="citation">(Kuleshov et al. 2016)</span> and <em>DAVID</em> <span class="citation">(Dennis et al. 2003)</span>, specialize in positioning novel findings against well curated data sources of biological processes and pathways. Most specifically, those tools identify functionally coherent gene sets that are statistically over- (or under-) represented in a gene list. The traditional output of enrichment analysis tool is a table containing the significant gene sets with their associated statistics. While those results are extremely useful, their interpretation is not always easy. In that respect, visual representation of the results as network could be handy.</p>
<p>Biological network models are visual representation of various biological interacting elements which are based on mathematical graphs. In those networks, the biological elements are generally represented by nodes while the interactions and relationships are represented by edges. One of the wildly used network tool in the quantitative biology community is the open source software <em>Cytoscape</em>. In addition of biological data visualization and network analysis, <em>Cytoscape</em> can be expended through the use of specialized plug-ins such as <em>BiNGO</em> that calculates over-represented GO terms in a network <span class="citation">(Maere, Heymans, and Kuiper 2005)</span> or <em>CentiScaPe</em> that identifies relevant network nodes <span class="citation">(Scardoni, Petterlini, and Laudanna 2009)</span>.</p>
<p>The <em>g:Profiler</em> enrichment analysis tool <span class="citation">(Raudvere et al. 2019)</span> is web based as the particularity of being accompanied by the CRAN package <em>gprofiler2</em> <span class="citation">(Kolberg et al. 2020)</span>. The <em>gprofiler2</em> package enables researchers to incorporate functional enrichment analysis into automated analysis pipelines written in R. This greatly facilitates research reproducibility.</p>
<p>The <em><a href="https://github.com/adeschen/gprofiler2cytoscape">gprofiler2cytoscape</a></em> package enables the visualization of enrichment results generated by <em>gprofiler2</em> as customisable <em>Cytoscape</em> network <span class="citation">(Shannon et al. 2003)</span>. In the biological networks generated by <em><a href="https://github.com/adeschen/gprofiler2cytoscape">gprofiler2cytoscape</a></em>, both gene datasets (GO terms/pathways/protein complexes) and genes associated to the datasets are represented as nodes. While the edges connect each gene to its dataset(s). Only genes present in the query used for the enrichment analysis are shown.</p>
<div class="figure" style="text-align: left">
<img src="demo01.jpeg" alt="Demo a biologocial network where GO terms and genes are presented as nodes while edges connect each gene to its dataset(s)." width="90%"><p class="caption">
Demo a biologocial network where GO terms and genes are presented as nodes while edges connect each gene to its dataset(s).
</p>
</div>
<p>Once the network is created, the user can personalize the visual attributes and integrate external information such as expression profiles, phenotypes and other molecular states. The user can also peforms network analysis.</p>
</div>
<div id="installation" class="section level1">
<h1 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h1>
<p>To install the latest version accessible on the <em><a href="https://github.com/adeschen/gprofiler2cytoscape">gprofiler2cytoscape</a></em> Github Website, the <em><a href="https://CRAN.R-project.org/package=devtools">devtools</a></em> package is required.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Load required package</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://devtools.r-lib.org/">devtools</a></span><span class="op">)</span>
<span class="co">## Install the latest version of gprofiler2cytoscape</span>
<span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">'adeschen/gprofiler2cytoscape'</span><span class="op">)</span></code></pre></div>
<p>It is also possible to install an official release. The list of available releases is posted on the <a href="https://github.com/adeschen/gprofiler2cytoscape/releases" title="gprofiler2cytoscape Release Site">gprofiler2cytoscape Release Website</a>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## Load required package</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://devtools.r-lib.org/">devtools</a></span><span class="op">)</span>
<span class="co">## Install the version v0.1 of gprofiler2cytoscape</span>
<span class="co">## using 'ref' parameter</span>
<span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">'adeschen/gprofiler2cytoscape'</span>, ref <span class="op">=</span> <span class="st">"v0.1"</span><span class="op">)</span></code></pre></div>
<p>TODO</p>
</div>
<div id="transforming-gprofiler2-enrichment-results-to-cytoscape-network" class="section level1">
<h1 class="hasAnchor">
<a href="#transforming-gprofiler2-enrichment-results-to-cytoscape-network" class="anchor"></a>Transforming gprofiler2 enrichment results to Cytoscape network</h1>
<div class="figure" style="text-align: left">
<img src="FromListToGraph_v01.jpg" alt="From an enrichment list to a Cytoscape network" width="100%"><p class="caption">
From an enrichment list to a Cytoscape network
</p>
</div>
</div>
<div id="session-info" class="section level1">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h1>
<p>Here is the output of sessionInfo() on the system on which this document was compiled:</p>
<pre><code>## R version 4.0.3 (2020-10-10)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 20.04 LTS
## 
## Matrix products: default
## BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] gprofiler2cytoscape_0.0.1 knitr_1.31               
## [3] BiocStyle_2.18.1         
## 
## loaded via a namespace (and not attached):
##  [1] tidyr_1.1.2         assertthat_0.2.1    rprojroot_2.0.2    
##  [4] digest_0.6.27       R6_2.5.0            stats4_4.0.3       
##  [7] evaluate_0.14       httr_1.4.2          ggplot2_3.3.3      
## [10] highr_0.8           pillar_1.4.7        rlang_0.4.10       
## [13] lazyeval_0.2.2      data.table_1.13.6   R.utils_2.10.1     
## [16] R.oo_1.24.0         rmarkdown_2.6       pkgdown_1.6.1      
## [19] textshaping_0.3.0   desc_1.2.0          stringr_1.4.0      
## [22] htmlwidgets_1.5.3   igraph_1.2.6        munsell_0.5.0      
## [25] compiler_4.0.3      gprofiler2_0.2.0    xfun_0.21          
## [28] pkgconfig_2.0.3     systemfonts_1.0.1   BiocGenerics_0.36.0
## [31] htmltools_0.5.1.1   tidyselect_1.1.0    tibble_3.0.6       
## [34] bookdown_0.21       XML_3.99-0.5        viridisLite_0.3.0  
## [37] crayon_1.4.1        dplyr_1.0.4         R.methodsS3_1.8.1  
## [40] grid_4.0.3          jsonlite_1.7.2      gtable_0.3.0       
## [43] lifecycle_1.0.0     magrittr_2.0.1      scales_1.1.1       
## [46] graph_1.68.0        stringi_1.5.3       cachem_1.0.4       
## [49] fs_1.5.0            ellipsis_0.3.1      ragg_1.1.0         
## [52] RCy3_2.10.2         vctrs_0.3.6         generics_0.1.0     
## [55] tools_4.0.3         RJSONIO_1.3-1.4     glue_1.4.2         
## [58] purrr_0.3.4         parallel_4.0.3      fastmap_1.1.0      
## [61] yaml_2.2.1          colorspace_2.0-0    BiocManager_1.30.10
## [64] strex_1.4.1         memoise_2.0.0       plotly_4.9.3</code></pre>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Dennis2003" class="csl-entry">
Dennis, Glynn, Brad T. Sherman, Douglas A. Hosack, Jun Yang, Wei Gao, H. Clifford Lane, and Richard A. Lempicki. 2003. <span>“<span class="nocase">DAVID: Database for Annotation, Visualization, and Integrated Discovery.</span>”</span> <em>Genome Biology</em> 4 (5). <a href="https://doi.org/10.1186/gb-2003-4-9-r60">https://doi.org/10.1186/gb-2003-4-9-r60</a>.
</div>
<div id="ref-Kolberg2020" class="csl-entry">
Kolberg, Liis, Uku Raudvere, Ivan Kuzmin, Jaak Vilo, and Hedi Peterson. 2020. <span>“<span class="nocase">gprofiler2 – an R package for gene list functional enrichment analysis and namespace conversion toolset g:Profiler</span>.”</span> <em>F1000Research</em> 9: 1–27. <a href="https://doi.org/10.12688/f1000research.24956.1">https://doi.org/10.12688/f1000research.24956.1</a>.
</div>
<div id="ref-Kuleshov2016" class="csl-entry">
Kuleshov, Maxim V., Matthew R. Jones, Andrew D. Rouillard, Nicolas F. Fernandez, Qiaonan Duan, Zichen Wang, Simon Koplev, et al. 2016. <span>“<span class="nocase">Enrichr: a comprehensive gene set enrichment analysis web server 2016 update</span>.”</span> <em>Nucleic Acids Research</em> 44 (W1): W90–97. <a href="https://doi.org/10.1093/nar/gkw377">https://doi.org/10.1093/nar/gkw377</a>.
</div>
<div id="ref-Maere2005" class="csl-entry">
Maere, Steven, Karel Heymans, and Martin Kuiper. 2005. <span>“<span class="nocase">BiNGO: a Cytoscape plugin to assess overrepresentation of Gene Ontology categories in Biological Networks</span>.”</span> <em>Bioinformatics</em> 21 (16): 3448–49. <a href="https://doi.org/10.1093/bioinformatics/bti551">https://doi.org/10.1093/bioinformatics/bti551</a>.
</div>
<div id="ref-Raudvere2019" class="csl-entry">
Raudvere, Uku, Liis Kolberg, Ivan Kuzmin, Tambet Arak, Priit Adler, Hedi Peterson, and Jaak Vilo. 2019. <span>“<span class="nocase">G:Profiler: A web server for functional enrichment analysis and conversions of gene lists (2019 update)</span>.”</span> <em>Nucleic Acids Research</em> 47 (W1): W191–98. <a href="https://doi.org/10.1093/nar/gkz369">https://doi.org/10.1093/nar/gkz369</a>.
</div>
<div id="ref-Scardoni2009" class="csl-entry">
Scardoni, Giovanni, Michele Petterlini, and Carlo Laudanna. 2009. <span>“<span class="nocase">Analyzing biological network parameters with CentiScaPe</span>.”</span> <em>Bioinformatics</em> 25 (21): 2857–59. <a href="https://doi.org/10.1093/bioinformatics/btp517">https://doi.org/10.1093/bioinformatics/btp517</a>.
</div>
<div id="ref-PaulShannon2003" class="csl-entry">
Shannon, Paul, Andrew Markiel, Owen Ozier, Nitin S Baliga, Jonathan T Wang, Daniel Ramage, Nada Amin, Benno Schwikowski, and Trey Ideker. 2003. <span>“<span class="nocase">Cytoscape: a software environment for integrated models of biomolecular interaction networks.</span>”</span> <em>Genome Research</em> 13 (11): 2498–2504. <a href="https://doi.org/10.1101/gr.1239303">https://doi.org/10.1101/gr.1239303</a>.
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Astrid Deschênes, Pascal Belleau.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
